// @coinbase/onchainkit/dist/minikit/index.js
import { M } from "../OnchainKitProvider-C29aWJbj.js";
import { useMiniKit } from "./hooks/useMiniKit.js";
import { useOpenUrl } from "./hooks/useOpenUrl.js";
import { parseSignInMessage, useAuthenticate } from "./hooks/useAuthenticate.js";
import { useViewProfile } from "./hooks/useViewProfile.js";
import { useAddFrame } from "./hooks/useAddFrame.js";
import { usePrimaryButton } from "./hooks/usePrimaryButton.js";
import { useNotification } from "./hooks/useNotification.js";
import { useClose } from "./hooks/useClose.js";
import { useComposeCast } from "./hooks/useComposeCast.js";
import { useViewCast } from "./hooks/useViewCast.js";
import { useIsInMiniApp } from "./hooks/useIsInMiniApp.js";
import { useSwapToken } from "./hooks/useSwapToken.js";
import { useSendToken } from "./hooks/useSendToken.js";
export {
  M as MiniKitProvider,
  parseSignInMessage,
  useAddFrame,
  useAuthenticate,
  useClose,
  useComposeCast,
  useIsInMiniApp,
  useMiniKit,
  useNotification,
  useOpenUrl,
  usePrimaryButton,
  useSendToken,
  useSwapToken,
  useViewCast,
  useViewProfile
};
