// @coinbase/onchainkit/dist/buy/components/BuyAmountInput.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { TextInput } from "../../internal/components/TextInput.js";
import { isValidAmount } from "../../internal/utils/isValidAmount.js";
import { cn, color, background, border } from "../../styles/theme.js";
import { formatAmount } from "../../swap/utils/formatAmount.js";
import { TokenChip } from "../../token/components/TokenChip.js";
import "react";
import "../../token/components/TokenRow.js";
import "../../useOnchainKit.js";
import "../../internal/svg/searchIconSvg.js";
import "react-dom";
import "../../token/components/TokenSelectButton.js";
import "viem";
import { useBuyContext } from "./BuyProvider.js";
function BuyAmountInput() {
  const { to, handleAmountChange } = useBuyContext();
  if (!(to == null ? void 0 : to.token)) {
    return null;
  }
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        "flex h-12 items-center border px-2 pl-4",
        background.default,
        border.radius,
        border.lineDefault
      ),
      children: [
        /* @__PURE__ */ jsx(
          TextInput,
          {
            className: cn(
              "mr-2 w-full border-none font-display",
              "leading-none outline-none disabled:cursor-not-allowed",
              background.default,
              color.foreground
            ),
            placeholder: "0.0",
            delayMs: 1e3,
            inputMode: "decimal",
            value: formatAmount(to.amount),
            setValue: to.setAmount,
            disabled: to.loading,
            onChange: handleAmountChange,
            inputValidator: isValidAmount
          }
        ),
        /* @__PURE__ */ jsx(
          TokenChip,
          {
            className: cn(color.foreground, "rounded-md"),
            token: to.token,
            isPressable: false
          }
        )
      ]
    }
  );
}
export {
  BuyAmountInput
};
