// @coinbase/onchainkit/dist/buy/components/BuyOnrampItem.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { usdcToken } from "../../token/constants.js";
import { useCallback, useMemo } from "react";
import { applePaySvg } from "../../internal/svg/applePaySvg.js";
import { cardSvg } from "../../internal/svg/cardSvg.js";
import { coinbaseLogoSvg } from "../../internal/svg/coinbaseLogoSvg.js";
import { cn, color, pressable, text } from "../../styles/theme.js";
import { useBuyContext } from "./BuyProvider.js";
const ONRAMP_ICON_MAP = {
  applePay: applePaySvg,
  coinbasePay: coinbaseLogoSvg,
  creditCard: cardSvg
};
function BuyOnrampItem({
  name,
  description,
  onClick,
  icon,
  to
}) {
  const { setIsDropdownOpen } = useBuyContext();
  const handleClick = useCallback(() => {
    setIsDropdownOpen(false);
    onClick();
  }, [onClick, setIsDropdownOpen]);
  const fiatAmount = useMemo(() => {
    var _a, _b, _c;
    if (((_b = (_a = to == null ? void 0 : to.token) == null ? void 0 : _a.address) == null ? void 0 : _b.toLowerCase()) === ((_c = usdcToken) == null ? void 0 : _c.address.toLowerCase())) {
      return to == null ? void 0 : to.amount;
    }
    return to == null ? void 0 : to.amountUSD;
  }, [to]);
  const isDisabled = !fiatAmount || Number.parseFloat(fiatAmount) < 5 && name !== "Coinbase";
  const message = useMemo(() => {
    if (isDisabled) {
      return "Minimum purchase amount is $5";
    }
    return description;
  }, [isDisabled, description]);
  return /* @__PURE__ */ jsxs(
    "button",
    {
      className: cn(
        "flex items-center gap-2 rounded-lg p-2",
        text.label2,
        !isDisabled && pressable.default,
        isDisabled && color.foregroundMuted
      ),
      onClick: handleClick,
      type: "button",
      "data-testid": `ock-${icon}OnrampItem`,
      disabled: isDisabled,
      children: [
        /* @__PURE__ */ jsx("div", { className: "flex h-9 w-9 items-center justify-center", children: ONRAMP_ICON_MAP[icon] }),
        /* @__PURE__ */ jsxs("div", { className: "flex flex-col items-start", children: [
          /* @__PURE__ */ jsx("div", { className: "relative flex items-center gap-1", children: /* @__PURE__ */ jsx("div", { children: name }) }),
          /* @__PURE__ */ jsx("div", { className: cn("text-xs", color.foregroundMuted), children: message })
        ] })
      ]
    }
  );
}
export {
  BuyOnrampItem
};
