// @coinbase/onchainkit/dist/buy/components/BuyTokenItem.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { useCallback, useMemo } from "react";
import { getRoundedAmount } from "../../internal/utils/getRoundedAmount.js";
import { cn, color, text, pressable } from "../../styles/theme.js";
import "../../useOnchainKit.js";
import { TokenImage } from "../../token/components/TokenImage.js";
import "../../token/components/TokenRow.js";
import "../../internal/components/TextInput.js";
import "../../internal/svg/searchIconSvg.js";
import "react-dom";
import "../../token/components/TokenSelectButton.js";
import "viem";
import { useBuyContext } from "./BuyProvider.js";
function BuyTokenItem({ swapUnit }) {
  const { handleSubmit, setIsDropdownOpen } = useBuyContext();
  if (!swapUnit || !swapUnit.token) {
    return null;
  }
  const handleClick = useCallback(() => {
    setIsDropdownOpen(false);
    handleSubmit(swapUnit);
  }, [handleSubmit, swapUnit, setIsDropdownOpen]);
  const hasInsufficientBalance = !swapUnit.balance || Number.parseFloat(swapUnit.balance) < Number.parseFloat(swapUnit.amount);
  const roundedAmount = useMemo(() => {
    if (!swapUnit.amount) {
      return "";
    }
    return getRoundedAmount(swapUnit.amount, 10);
  }, [swapUnit.amount]);
  const roundedBalance = useMemo(() => {
    return getRoundedAmount(swapUnit.balance || "0", 3);
  }, [swapUnit.balance]);
  return /* @__PURE__ */ jsxs(
    "button",
    {
      className: cn(
        "flex items-center gap-2 rounded-lg p-2",
        !hasInsufficientBalance && pressable.default,
        text.label2
      ),
      onClick: handleClick,
      type: "button",
      disabled: hasInsufficientBalance,
      children: [
        /* @__PURE__ */ jsx(TokenImage, { token: swapUnit.token, size: 36 }),
        /* @__PURE__ */ jsxs(
          "div",
          {
            className: cn(
              "flex flex-col items-start",
              hasInsufficientBalance ? color.foregroundMuted : color.foreground
            ),
            children: [
              /* @__PURE__ */ jsxs("div", { children: [
                roundedAmount,
                " ",
                swapUnit.token.name
              ] }),
              /* @__PURE__ */ jsx(
                "div",
                {
                  className: cn(
                    "text-xs",
                    hasInsufficientBalance ? color.error : color.foregroundMuted
                  ),
                  children: `${hasInsufficientBalance ? "Insufficient balance" : "Balance"}: ${roundedBalance}`
                }
              )
            ]
          }
        )
      ]
    }
  );
}
export {
  BuyTokenItem
};
