// @coinbase/onchainkit/dist/swap/components/SwapToast.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { useCallback } from "react";
import { cn, text, color } from "../../styles/theme.js";
import { useAccount } from "wagmi";
import { getChainExplorer } from "../../core/network/getChainExplorer.js";
import { Toast } from "../../internal/components/Toast.js";
import { SuccessSvg } from "../../internal/svg/successSvg.js";
import { useSwapContext } from "./SwapProvider.js";
function SwapToast({
  className,
  durationMs = 5e3,
  position = "bottom-center"
}) {
  const {
    isToastVisible,
    setIsToastVisible,
    setTransactionHash,
    transactionHash
  } = useSwapContext();
  const { chainId } = useAccount();
  const chainExplorer = getChainExplorer(chainId);
  const resetToastState = useCallback(() => {
    setIsToastVisible(false);
    setTransactionHash("");
  }, [setIsToastVisible, setTransactionHash]);
  if (!isToastVisible) {
    return null;
  }
  return /* @__PURE__ */ jsxs(
    Toast,
    {
      position,
      className,
      durationMs,
      isVisible: isToastVisible,
      onClose: resetToastState,
      children: [
        /* @__PURE__ */ jsx("div", { className: cn(text.label2), children: /* @__PURE__ */ jsx(SuccessSvg, {}) }),
        /* @__PURE__ */ jsx("div", { className: cn(text.label1, "text-nowrap"), children: /* @__PURE__ */ jsx("p", { className: color.foreground, children: "Successful" }) }),
        /* @__PURE__ */ jsx("div", { className: cn(text.label1, "text-nowrap"), children: /* @__PURE__ */ jsx(
          "a",
          {
            href: `${chainExplorer}/tx/${transactionHash}`,
            target: "_blank",
            rel: "noreferrer",
            children: /* @__PURE__ */ jsx("span", { className: cn(text.label1, color.primary), children: "View transaction" })
          }
        ) })
      ]
    }
  );
}
export {
  SwapToast
};
