// @coinbase/onchainkit/dist/swap/components/SwapButton.js
'use client';
import { jsx } from "react/jsx-runtime";
import { Spinner } from "../../internal/components/Spinner.js";
import { cn, color, text, pressable, border, background } from "../../styles/theme.js";
import { ConnectWallet } from "../../wallet/components/ConnectWallet.js";
import { useSwapContext } from "./SwapProvider.js";
function SwapButton({
  className,
  label = "Swap",
  disabled = false
}) {
  var _a, _b;
  const {
    address,
    to,
    from,
    lifecycleStatus: { statusName },
    handleSubmit
  } = useSwapContext();
  const isLoading = to.loading || from.loading || statusName === "transactionPending" || statusName === "transactionApproved";
  const isDisabled = !from.amount || !from.token || !to.amount || !to.token || disabled || isLoading;
  const isSwapInvalid = ((_a = to.token) == null ? void 0 : _a.address) === ((_b = from.token) == null ? void 0 : _b.address);
  if (!isDisabled && !address) {
    return /* @__PURE__ */ jsx(ConnectWallet, { className: cn("mt-4 w-full", className) });
  }
  return /* @__PURE__ */ jsx(
    "button",
    {
      type: "button",
      className: cn(
        background.primary,
        border.radius,
        "w-full rounded-xl",
        "mt-4 px-4 py-3",
        isDisabled && pressable.disabled,
        text.headline,
        className
      ),
      onClick: () => handleSubmit(),
      disabled: isDisabled || isSwapInvalid,
      "data-testid": "ockSwapButton_Button",
      children: isLoading ? /* @__PURE__ */ jsx(Spinner, {}) : /* @__PURE__ */ jsx("span", { className: cn(text.headline, color.inverse), children: label })
    }
  );
}
export {
  SwapButton
};
