// @coinbase/onchainkit/dist/swap/components/SwapSettings.js
import { jsxs, jsx, Fragment } from "react/jsx-runtime";
import { DismissableLayer } from "../../internal/components/DismissableLayer.js";
import { FocusTrap } from "../../internal/components/FocusTrap.js";
import { Popover } from "../../internal/components/Popover.js";
import { useBreakpoints } from "../../internal/hooks/useBreakpoints.js";
import { useIcon } from "../../internal/hooks/useIcon.js";
import { cn, text, pressable, background, border } from "../../styles/theme.js";
import { useState, useRef, useCallback } from "react";
import { SwapSettingsSlippageDescription } from "./SwapSettingsSlippageDescription.js";
import { SwapSettingsSlippageInput } from "./SwapSettingsSlippageInput.js";
import { SwapSettingsSlippageLayout } from "./SwapSettingsSlippageLayout.js";
import { SwapSettingsSlippageLayoutBottomSheet } from "./SwapSettingsSlippageLayoutBottomSheet.js";
import { SwapSettingsSlippageTitle } from "./SwapSettingsSlippageTitle.js";
const DEFAULT_CHILDREN = /* @__PURE__ */ jsxs(Fragment, { children: [
  /* @__PURE__ */ jsx(SwapSettingsSlippageTitle, { children: "Max. slippage" }),
  /* @__PURE__ */ jsx(SwapSettingsSlippageDescription, { children: "Your swap will revert if the prices change by more than the selected percentage." }),
  /* @__PURE__ */ jsx(SwapSettingsSlippageInput, {})
] });
function SwapSettings({
  children = DEFAULT_CHILDREN,
  className,
  icon = "swapSettings",
  text: buttonText = ""
}) {
  const breakpoint = useBreakpoints();
  const [isOpen, setIsOpen] = useState(false);
  const dropdownRef = useRef(null);
  const triggerRef = useRef(null);
  const handleToggle = useCallback((e) => {
    e.stopPropagation();
    setIsOpen((prev) => !prev);
  }, []);
  const handleClose = useCallback(() => {
    setIsOpen(false);
  }, []);
  const iconSvg = useIcon({ icon });
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        "flex w-auto items-center justify-end space-x-1 pb-4",
        className
      ),
      "data-testid": "ockSwapSettings_Settings",
      children: [
        buttonText && /* @__PURE__ */ jsx("span", { className: cn(text.body), children: buttonText }),
        /* @__PURE__ */ jsxs("div", { className: cn("relative", isOpen && "group"), ref: dropdownRef, children: [
          /* @__PURE__ */ jsx(
            "button",
            {
              ref: triggerRef,
              type: "button",
              "aria-label": "Toggle swap settings",
              className: cn(
                pressable.default,
                "rounded-full p-2 opacity-50 transition-opacity hover:opacity-100"
              ),
              onClick: handleToggle,
              children: /* @__PURE__ */ jsx("div", { className: "h-[1.125rem] w-[1.125rem]", children: iconSvg })
            }
          ),
          breakpoint === "sm" ? /* @__PURE__ */ jsx(FocusTrap, { active: isOpen, children: /* @__PURE__ */ jsx(
            DismissableLayer,
            {
              onDismiss: handleClose,
              triggerRef,
              preventTriggerEvents: true,
              children: /* @__PURE__ */ jsx(
                "div",
                {
                  className: cn(
                    background.inverse,
                    pressable.shadow,
                    "fixed inset-x-0 z-50 transition-[bottom] duration-300 ease-in-out",
                    isOpen ? "bottom-0" : "-bottom-[12.875rem]",
                    "h-[12.875rem] rounded-t-lg",
                    className
                  ),
                  "data-testid": "ockSwapSettingsSlippageLayoutBottomSheet_container",
                  children: /* @__PURE__ */ jsx(SwapSettingsSlippageLayoutBottomSheet, { className, children })
                }
              )
            }
          ) }) : /* @__PURE__ */ jsx(
            Popover,
            {
              isOpen,
              onClose: handleClose,
              anchor: dropdownRef.current,
              position: "bottom",
              align: "end",
              trigger: triggerRef,
              children: /* @__PURE__ */ jsx(
                "div",
                {
                  className: cn(
                    border.radius,
                    background.default,
                    pressable.shadow,
                    "absolute right-0 z-10 mt-1 w-[21.75rem] rounded-lg"
                  ),
                  "data-testid": "ockSwapSettingsDropdown",
                  children: /* @__PURE__ */ jsx(SwapSettingsSlippageLayout, { children })
                }
              )
            }
          )
        ] })
      ]
    }
  );
}
export {
  SwapSettings
};
