// @coinbase/onchainkit/dist/swap/components/SwapSettingsSlippageInput.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { useAnalytics } from "../../core/analytics/hooks/useAnalytics.js";
import { SwapEvent } from "../../core/analytics/types.js";
import { useState, useCallback } from "react";
import { cn, color, background, pressable, border, text } from "../../styles/theme.js";
import { useSwapContext } from "./SwapProvider.js";
const SLIPPAGE_SETTINGS = {
  AUTO: "Auto",
  CUSTOM: "Custom"
};
function SwapSettingsSlippageInput({
  className
}) {
  const { sendAnalytics } = useAnalytics();
  const {
    config: { maxSlippage: defaultMaxSlippage },
    updateLifecycleStatus,
    lifecycleStatus
  } = useSwapContext();
  const [slippageSetting, setSlippageSetting] = useState(
    lifecycleStatus.statusData.maxSlippage === defaultMaxSlippage ? SLIPPAGE_SETTINGS.AUTO : SLIPPAGE_SETTINGS.CUSTOM
  );
  const handleAnalyticsSlippageChange = useCallback(
    (previousSlippage, newSlippage) => {
      sendAnalytics(SwapEvent.SlippageChanged, {
        previousSlippage,
        slippage: newSlippage
      });
    },
    [sendAnalytics]
  );
  const updateSlippage = useCallback(
    (newSlippage) => {
      const currentSlippage = lifecycleStatus.statusData.maxSlippage;
      if (newSlippage !== currentSlippage) {
        handleAnalyticsSlippageChange(currentSlippage, newSlippage);
        updateLifecycleStatus({
          statusName: "slippageChange",
          statusData: {
            maxSlippage: newSlippage
          }
        });
      }
    },
    [
      lifecycleStatus.statusData.maxSlippage,
      updateLifecycleStatus,
      handleAnalyticsSlippageChange
    ]
  );
  const handleSlippageChange = useCallback(
    (e) => {
      const newSlippage = e.target.value;
      const parsedSlippage = Number.parseFloat(newSlippage);
      const isValidNumber = !Number.isNaN(parsedSlippage);
      updateSlippage(isValidNumber ? parsedSlippage : 0);
    },
    [updateSlippage]
  );
  const handleSlippageSettingChange = useCallback(
    (setting) => {
      setSlippageSetting(setting);
      if (setting === SLIPPAGE_SETTINGS.AUTO) {
        updateSlippage(defaultMaxSlippage);
      }
    },
    [defaultMaxSlippage, updateSlippage]
  );
  return /* @__PURE__ */ jsxs(
    "section",
    {
      className: cn(
        background.default,
        border.defaultActive,
        border.radius,
        "flex items-center gap-2 flex-grow max-sm:pt-4",
        className
      ),
      children: [
        /* @__PURE__ */ jsxs(
          "fieldset",
          {
            className: cn(
              background.default,
              border.defaultActive,
              border.radius,
              "flex h-9 flex-1 rounded-xl border p-1"
            ),
            children: [
              /* @__PURE__ */ jsx("legend", { className: "sr-only", children: "Slippage Setting" }),
              Object.values(SLIPPAGE_SETTINGS).map((setting) => /* @__PURE__ */ jsx(
                "button",
                {
                  type: "button",
                  className: cn(
                    pressable.default,
                    color.foreground,
                    text.label1,
                    border.radiusInner,
                    "flex-1 px-3 py-1 transition-colors",
                    // Highlight the button if it is selected
                    slippageSetting === setting ? cn(background.inverse, color.primary, pressable.shadow) : color.foregroundMuted
                  ),
                  onClick: () => handleSlippageSettingChange(setting),
                  children: setting
                },
                setting
              ))
            ]
          }
        ),
        /* @__PURE__ */ jsxs(
          "div",
          {
            className: cn(
              background.default,
              border.defaultActive,
              border.radius,
              "flex h-9 w-24 items-center justify-between border px-2 py-1",
              slippageSetting === SLIPPAGE_SETTINGS.AUTO && "opacity-50"
            ),
            children: [
              /* @__PURE__ */ jsx("label", { htmlFor: "slippage-input", className: "sr-only", children: "Slippage Percentage" }),
              /* @__PURE__ */ jsx(
                "input",
                {
                  id: "slippage-input",
                  type: "text",
                  value: lifecycleStatus.statusData.maxSlippage,
                  onChange: handleSlippageChange,
                  disabled: slippageSetting === SLIPPAGE_SETTINGS.AUTO,
                  className: cn(
                    color.foreground,
                    text.label2,
                    "w-full flex-grow bg-transparent pl-1 font-normal leading-6 focus:outline-none",
                    slippageSetting === SLIPPAGE_SETTINGS.AUTO && "cursor-not-allowed"
                  )
                }
              ),
              /* @__PURE__ */ jsx(
                "span",
                {
                  className: cn(
                    background.default,
                    color.foreground,
                    text.label2,
                    "ml-1 flex-shrink-0 font-normal leading-6"
                  ),
                  children: "%"
                }
              )
            ]
          }
        )
      ]
    }
  );
}
export {
  SwapSettingsSlippageInput
};
