// @coinbase/onchainkit/dist/nft/components/mint/NFTTotalCost.js
import { jsxs, jsx } from "react/jsx-runtime";
import { infoSvg } from "../../../internal/svg/infoSvg.js";
import { multiplyFloats } from "../../../internal/utils/multiplyFloats.js";
import { useNFTContext } from "../NFTProvider.js";
import { cn, text, border, background } from "../../../styles/theme.js";
import { formatAmount } from "../../../token/utils/formatAmount.js";
import { useState, useCallback, useMemo } from "react";
function NFTTotalCost({
  className,
  label = "Total cost"
}) {
  const [isOverlayVisible, setIsOverlayVisible] = useState(false);
  const { price, mintFee, quantity } = useNFTContext();
  const toggleOverlay = useCallback(() => {
    setIsOverlayVisible((prev) => !prev);
  }, []);
  const showOverlay = useCallback(() => {
    setIsOverlayVisible(true);
  }, []);
  const hideOverlay = useCallback(() => {
    setIsOverlayVisible(false);
  }, []);
  const overlay = useMemo(() => {
    if ((price == null ? void 0 : price.amount) === void 0 || (price == null ? void 0 : price.amountUSD) === void 0 || (mintFee == null ? void 0 : mintFee.amount) === void 0 || mintFee.amountUSD === void 0) {
      return null;
    }
    return /* @__PURE__ */ jsxs(
      "div",
      {
        className: cn(
          background.default,
          border.radius,
          border.defaultActive,
          "absolute z-10 w-full border"
        ),
        children: [
          /* @__PURE__ */ jsxs(
            "div",
            {
              className: cn(
                text.label2,
                "flex items-center justify-between px-4 py-2"
              ),
              children: [
                /* @__PURE__ */ jsx("div", { children: "NFT cost" }),
                /* @__PURE__ */ jsxs("div", { children: [
                  "$",
                  formatAmount(
                    `${multiplyFloats(Number(price.amountUSD), quantity)}`,
                    {
                      minimumFractionDigits: 2,
                      maximumFractionDigits: 2
                    }
                  )
                ] })
              ]
            }
          ),
          /* @__PURE__ */ jsxs(
            "div",
            {
              className: cn(
                "flex items-center justify-between px-4 py-2",
                text.label2
              ),
              children: [
                /* @__PURE__ */ jsx("div", { children: "Mint fee" }),
                /* @__PURE__ */ jsxs("div", { children: [
                  "$",
                  formatAmount(
                    `${multiplyFloats(Number(mintFee.amountUSD), quantity)}`,
                    {
                      minimumFractionDigits: 2,
                      maximumFractionDigits: 2
                    }
                  )
                ] })
              ]
            }
          )
        ]
      }
    );
  }, [mintFee, price, quantity]);
  if (!(price == null ? void 0 : price.amount) || !(price == null ? void 0 : price.currency) || !(price == null ? void 0 : price.amountUSD) || !(mintFee == null ? void 0 : mintFee.amount) || !mintFee.amountUSD) {
    return null;
  }
  return /* @__PURE__ */ jsxs("div", { className: "relative", children: [
    /* @__PURE__ */ jsxs(
      "div",
      {
        className: cn(
          text.label2,
          "flex items-center justify-between",
          className
        ),
        children: [
          /* @__PURE__ */ jsx("div", { children: label }),
          /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-2", children: [
            /* @__PURE__ */ jsxs("div", { children: [
              "$",
              formatAmount(
                `${multiplyFloats(Number(price.amountUSD), quantity)}`,
                {
                  minimumFractionDigits: 2,
                  maximumFractionDigits: 2
                }
              )
            ] }),
            overlay && /* @__PURE__ */ jsx(
              "button",
              {
                type: "button",
                "data-testid": "ockNFTTotalCostInfo",
                className: "h-2.5 w-2.5 cursor-pointer object-cover",
                onClick: toggleOverlay,
                onMouseEnter: showOverlay,
                onMouseLeave: hideOverlay,
                children: infoSvg
              }
            )
          ] })
        ]
      }
    ),
    isOverlayVisible && overlay
  ] });
}
export {
  NFTTotalCost
};
