// @coinbase/onchainkit/dist/internal/components/Dialog.js
import { jsx } from "react/jsx-runtime";
import { useTheme } from "../hooks/useTheme.js";
import { zIndex } from "../../styles/constants.js";
import { cn } from "../../styles/theme.js";
import { useRef } from "react";
import { createPortal } from "react-dom";
import { DismissableLayer } from "./DismissableLayer.js";
import { FocusTrap } from "./FocusTrap.js";
function Dialog({
  children,
  isOpen,
  modal = true,
  onClose,
  "aria-label": ariaLabel,
  "aria-labelledby": ariaLabelledby,
  "aria-describedby": ariaDescribedby
}) {
  const componentTheme = useTheme();
  const dialogRef = useRef(null);
  if (!isOpen) {
    return null;
  }
  const dialog = /* @__PURE__ */ jsx(
    "div",
    {
      className: cn(
        componentTheme,
        zIndex.modal,
        "fixed inset-0 flex items-center justify-center",
        "bg-black/50 transition-opacity duration-200",
        "fade-in animate-in duration-200"
      ),
      "data-portal-origin": "true",
      children: /* @__PURE__ */ jsx(FocusTrap, { active: isOpen, children: /* @__PURE__ */ jsx(DismissableLayer, { onDismiss: onClose, children: /* @__PURE__ */ jsx(
        "div",
        {
          "aria-describedby": ariaDescribedby,
          "aria-label": ariaLabel,
          "aria-labelledby": ariaLabelledby,
          "aria-modal": modal,
          className: "zoom-in-95 animate-in duration-200",
          "data-testid": "ockDialog",
          onClick: (e) => e.stopPropagation(),
          onKeyDown: (e) => {
            if (e.key === "Enter" || e.key === " ") {
              e.stopPropagation();
            }
          },
          ref: dialogRef,
          role: "dialog",
          children
        }
      ) }) })
    }
  );
  return createPortal(dialog, document.body);
}
export {
  Dialog
};
