// @coinbase/onchainkit/dist/wallet/components/wallet-advanced-send/components/SendAddressInput.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { TextInput } from "../../../../internal/components/TextInput.js";
import { cn, color, background, border } from "../../../../styles/theme.js";
import { useSendContext } from "./SendProvider.js";
function SendAddressInput({ classNames }) {
  const {
    recipientState,
    updateRecipientInput,
    validateRecipientInput,
    deselectRecipient
  } = useSendContext();
  return /* @__PURE__ */ jsxs(
    "div",
    {
      "data-testid": "ockSendAddressInput",
      className: cn(
        border.lineDefault,
        border.radius,
        "w-full",
        "flex items-center gap-2",
        "px-4 py-3",
        classNames == null ? void 0 : classNames.container
      ),
      children: [
        /* @__PURE__ */ jsx("span", { className: cn(color.foreground, classNames == null ? void 0 : classNames.label), children: "To" }),
        /* @__PURE__ */ jsx(
          TextInput,
          {
            inputMode: "text",
            placeholder: "Basename, ENS, or Address",
            value: recipientState.displayValue ?? recipientState.input,
            setValue: updateRecipientInput,
            onChange: validateRecipientInput,
            onFocus: deselectRecipient,
            "aria-label": "Input Receiver Address",
            className: cn(
              background.default,
              "w-full outline-none",
              classNames == null ? void 0 : classNames.input
            )
          }
        )
      ]
    }
  );
}
export {
  SendAddressInput
};
