// @coinbase/onchainkit/dist/wallet/components/wallet-advanced-send/components/SendTokenSelector.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { cn, color, text, border, pressable } from "../../../../styles/theme.js";
import "../../../../useOnchainKit.js";
import "react";
import "../../../../token/components/TokenRow.js";
import "../../../../internal/components/TextInput.js";
import "../../../../internal/svg/searchIconSvg.js";
import "react-dom";
import "../../../../token/components/TokenSelectButton.js";
import { TokenBalance } from "../../../../token/components/TokenBalance.js";
import { formatUnits } from "viem";
import { useSendContext } from "./SendProvider.js";
import { RequestContext } from "../../../../core/network/constants.js";
import { useAccount } from "wagmi";
import { usePortfolio } from "../../../hooks/usePortfolio.js";
function SendTokenSelector({ classNames }) {
  const { address } = useAccount();
  const { data: portfolioData } = usePortfolio(
    { address },
    RequestContext.Wallet
  );
  const tokenBalances = portfolioData == null ? void 0 : portfolioData.tokenBalances;
  const {
    selectedToken,
    handleTokenSelection,
    handleResetTokenSelection,
    setSelectedInputType,
    handleCryptoAmountChange,
    handleFiatAmountChange
  } = useSendContext();
  if (!selectedToken) {
    return /* @__PURE__ */ jsxs("div", { className: "mt-4 flex max-h-80 flex-col gap-2", children: [
      /* @__PURE__ */ jsx("span", { className: cn(text.caption, color.foregroundMuted, "uppercase"), children: "Select a token" }),
      /* @__PURE__ */ jsx("div", { className: "scrollbar-hidden overflow-y-auto", children: tokenBalances == null ? void 0 : tokenBalances.map((token) => /* @__PURE__ */ jsx(
        TokenBalance,
        {
          token,
          onClick: handleTokenSelection,
          subtitle: "",
          classNames: {
            container: cn(
              pressable.default,
              border.radius,
              classNames == null ? void 0 : classNames.container
            ),
            ...classNames
          }
        },
        token.address
      )) })
    ] });
  }
  return /* @__PURE__ */ jsx(
    TokenBalance,
    {
      token: selectedToken,
      showImage: true,
      subtitle: "available",
      onClick: handleResetTokenSelection,
      onActionPress: () => {
        setSelectedInputType("crypto");
        handleFiatAmountChange(String(selectedToken.fiatBalance));
        handleCryptoAmountChange(
          String(
            formatUnits(
              BigInt(selectedToken.cryptoBalance),
              selectedToken.decimals
            )
          )
        );
      },
      classNames: {
        container: cn(
          pressable.alternate,
          border.radius,
          classNames == null ? void 0 : classNames.container
        ),
        ...classNames
      }
    }
  );
}
export {
  SendTokenSelector
};
