// @coinbase/onchainkit/dist/wallet/components/wallet-advanced-send/components/SendAddressSelector.js
'use client';
import { jsx, jsxs } from "react/jsx-runtime";
import { Address } from "../../../../identity/components/Address.js";
import { Avatar } from "../../../../identity/components/Avatar.js";
import "../../../../core/network/attestations.js";
import "../../../../identity/utils/easSupportedChains.js";
import "@tanstack/react-query";
import "../../../../internal/svg/badgeSvg.js";
import { cn, pressable, border, background } from "../../../../styles/theme.js";
import { useCallback } from "react";
import "../../../../useOnchainKit.js";
import "../../../../identity/components/IdentityProvider.js";
import "viem";
import "wagmi";
import { Name } from "../../../../identity/components/Name.js";
import "viem/chains";
import "../../../../identity/constants.js";
import "../../../../identity/utils/getAddress.js";
import "viem/ens";
import "../../../../internal/svg/githubSvg.js";
import "../../../../internal/svg/twitterSvg.js";
import "../../../../internal/svg/warpcastSvg.js";
import "../../../../internal/svg/websiteSvg.js";
import "../../../../identity/utils/getAddresses.js";
import { useWalletContext } from "../../WalletProvider.js";
import { useSendContext } from "./SendProvider.js";
function SendAddressSelector({ classNames }) {
  const { chain } = useWalletContext();
  const { recipientState, selectRecipient } = useSendContext();
  const handleSelectorClick = useCallback(async () => {
    selectRecipient({
      phase: "selected",
      input: recipientState.input,
      address: recipientState.address,
      displayValue: recipientState.displayValue
    });
  }, [
    recipientState.input,
    recipientState.address,
    recipientState.displayValue,
    selectRecipient
  ]);
  if (!recipientState.address || !chain) {
    return null;
  }
  return /* @__PURE__ */ jsx(
    "button",
    {
      "data-testid": "ockSendAddressSelector_button",
      type: "button",
      onClick: handleSelectorClick,
      className: "w-full text-left",
      children: /* @__PURE__ */ jsx(
        "div",
        {
          "data-testid": "ockSendAddressSelector_container",
          className: cn(
            background.default,
            border.radius,
            pressable.default,
            "items-left flex min-w-[300px]",
            "mt-2 p-2",
            classNames == null ? void 0 : classNames.container
          ),
          children: /* @__PURE__ */ jsxs("div", { className: "flex items-center space-x-3", children: [
            /* @__PURE__ */ jsx("div", { className: "flex-shrink-0", children: /* @__PURE__ */ jsx(
              Avatar,
              {
                address: recipientState.address,
                chain,
                className: classNames == null ? void 0 : classNames.avatar
              }
            ) }),
            /* @__PURE__ */ jsxs("div", { className: "flex flex-col", children: [
              /* @__PURE__ */ jsx(
                Name,
                {
                  address: recipientState.address,
                  chain,
                  className: classNames == null ? void 0 : classNames.name
                }
              ),
              /* @__PURE__ */ jsx(
                Address,
                {
                  address: recipientState.address,
                  hasCopyAddressOnClick: false,
                  className: classNames == null ? void 0 : classNames.address
                }
              )
            ] })
          ] })
        }
      )
    }
  );
}
export {
  SendAddressSelector
};
