// @coinbase/onchainkit/dist/wallet/components/wallet-advanced-send/components/SendAmountInputTypeSwitch.js
import { jsx } from "react/jsx-runtime";
import { Skeleton } from "../../../../internal/components/Skeleton.js";
import { AmountInputTypeSwitch } from "../../../../internal/components/amount-input/AmountInputTypeSwitch.js";
import { cn, color, text } from "../../../../styles/theme.js";
import { useSendContext } from "./SendProvider.js";
const defaultErrorDisplay = /* @__PURE__ */ jsx(
  "div",
  {
    "data-testid": "ockSendAmountInputTypeSwitch_ErrorDisplay",
    className: cn(text.caption, color.foregroundMuted, "h-[1.625rem]"),
    children: "Exchange rate unavailable"
  }
);
function SendAmountInputTypeSwitch({
  errorDisplay,
  className
}) {
  const {
    selectedToken,
    fiatAmount,
    cryptoAmount,
    exchangeRate,
    exchangeRateLoading,
    selectedInputType,
    setSelectedInputType
  } = useSendContext();
  if (exchangeRateLoading) {
    return /* @__PURE__ */ jsx(Skeleton, { className: "h-[1.625rem]" });
  }
  if (!exchangeRate) {
    if (errorDisplay) {
      return errorDisplay;
    }
    return defaultErrorDisplay;
  }
  return /* @__PURE__ */ jsx(
    AmountInputTypeSwitch,
    {
      asset: (selectedToken == null ? void 0 : selectedToken.symbol) ?? "",
      fiatAmount: fiatAmount ?? "",
      cryptoAmount: cryptoAmount ?? "",
      exchangeRate,
      exchangeRateLoading: false,
      currency: "USD",
      selectedInputType,
      setSelectedInputType,
      className
    }
  );
}
export {
  SendAmountInputTypeSwitch
};
