// @coinbase/onchainkit/dist/wallet/components/wallet-advanced-send/components/SendFundWallet.js
import { jsxs, jsx } from "react/jsx-runtime";
import "react";
import "../../../../useOnchainKit.js";
import { cn, color, text } from "../../../../styles/theme.js";
import "wagmi";
import "viem/chains";
import "@farcaster/frame-sdk";
import "../../../../identity/components/IdentityProvider.js";
import "viem";
import "viem/ens";
import "../../../../identity/constants.js";
import "@tanstack/react-query";
import "../../../../identity/utils/getAddress.js";
import "../../../../internal/svg/defaultAvatarSVG.js";
import "../../../../core/network/attestations.js";
import "../../../../identity/utils/easSupportedChains.js";
import "../../../../internal/svg/badgeSvg.js";
import "../../../../internal/svg/githubSvg.js";
import "../../../../internal/svg/twitterSvg.js";
import "../../../../internal/svg/warpcastSvg.js";
import "../../../../internal/svg/websiteSvg.js";
import "../../../../identity/utils/getAddresses.js";
import "react-dom";
import "../../../../internal/svg/trustWalletSvg.js";
import "wagmi/connectors";
import "../../WalletProvider.js";
import "wagmi/experimental";
import { FundCard } from "../../../../fund/components/FundCard.js";
import { FundCardAmountInput } from "../../../../fund/components/FundCardAmountInput.js";
import { FundCardAmountInputTypeSwitch } from "../../../../fund/components/FundCardAmountInputTypeSwitch.js";
import "../../../../fund/components/FundCardProvider.js";
import { FundCardPaymentMethodDropdown } from "../../../../fund/components/FundCardPaymentMethodDropdown.js";
import { FundCardPresetAmountInputList } from "../../../../fund/components/FundCardPresetAmountInputList.js";
import { FundCardSubmitButton } from "../../../../fund/components/FundCardSubmitButton.js";
function SendFundWallet({
  onError,
  onStatus,
  onSuccess,
  classNames
}) {
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        "flex flex-col items-center justify-between",
        classNames == null ? void 0 : classNames.container
      ),
      "data-testid": "ockSendFundWallet",
      children: [
        /* @__PURE__ */ jsx(
          "div",
          {
            className: cn(text.label2, color.foregroundMuted, classNames == null ? void 0 : classNames.subtitle),
            children: "Insufficient ETH balance to send transaction. Fund your wallet to continue."
          }
        ),
        /* @__PURE__ */ jsxs(
          FundCard,
          {
            assetSymbol: "ETH",
            country: "US",
            currency: "USD",
            presetAmountInputs: ["2", "5", "10"],
            onError,
            onStatus,
            onSuccess,
            className: cn("mt-3 w-88 border-none py-0", classNames == null ? void 0 : classNames.fundCard),
            children: [
              /* @__PURE__ */ jsx(FundCardAmountInput, {}),
              /* @__PURE__ */ jsx(FundCardAmountInputTypeSwitch, {}),
              /* @__PURE__ */ jsx(FundCardPresetAmountInputList, {}),
              /* @__PURE__ */ jsx(FundCardPaymentMethodDropdown, {}),
              /* @__PURE__ */ jsx(FundCardSubmitButton, {})
            ]
          }
        )
      ]
    }
  );
}
export {
  SendFundWallet
};
