// @coinbase/onchainkit/dist/token/components/TokenBalance.js
import { jsxs, jsx } from "react/jsx-runtime";
import { formatFiatAmount } from "../../internal/utils/formatFiatAmount.js";
import { truncateDecimalPlaces } from "../../internal/utils/truncateDecimalPlaces.js";
import { cn, color, text, border } from "../../styles/theme.js";
import { TokenImage } from "./TokenImage.js";
import { useMemo } from "react";
import { formatUnits } from "viem";
function TokenBalance({
  token,
  onClick,
  onActionPress,
  actionText = "Max",
  classNames,
  "aria-label": ariaLabel,
  ...contentProps
}) {
  if (onClick) {
    return /* @__PURE__ */ jsxs("div", { className: "relative", children: [
      /* @__PURE__ */ jsx(
        "button",
        {
          type: "button",
          "aria-label": ariaLabel ?? `${token.name} token balance`,
          onClick: () => onClick(token),
          className: cn(
            "flex w-full items-center justify-start gap-4 px-2 py-1",
            classNames == null ? void 0 : classNames.container
          ),
          "data-testid": "ockTokenBalanceButton",
          children: /* @__PURE__ */ jsx(
            TokenBalanceContent,
            {
              token,
              classNames,
              onActionPress,
              ...contentProps
            }
          )
        }
      ),
      onActionPress && /* @__PURE__ */ jsx(
        ActionButton,
        {
          actionText,
          onActionPress,
          className: classNames == null ? void 0 : classNames.action
        }
      )
    ] });
  }
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        "flex w-full items-center justify-start gap-4 px-2 py-1",
        classNames == null ? void 0 : classNames.container
      ),
      "data-testid": "ockTokenBalance",
      children: [
        /* @__PURE__ */ jsx(
          TokenBalanceContent,
          {
            token,
            ...contentProps,
            classNames
          }
        ),
        onActionPress && /* @__PURE__ */ jsx(
          ActionButton,
          {
            actionText,
            onActionPress,
            className: classNames == null ? void 0 : classNames.action
          }
        )
      ]
    }
  );
}
function TokenBalanceContent({
  token,
  subtitle,
  showImage = true,
  onActionPress,
  tokenSize = 40,
  classNames
}) {
  var _a;
  const formattedFiatValue = useMemo(
    () => formatFiatAmount({
      amount: token.fiatBalance,
      currency: "USD"
    }),
    [token.fiatBalance]
  );
  const formattedCryptoValue = useMemo(
    () => truncateDecimalPlaces(
      formatUnits(BigInt(token.cryptoBalance), token.decimals),
      3
    ),
    [token.cryptoBalance, token.decimals]
  );
  return /* @__PURE__ */ jsxs("div", { className: "grid w-full grid-cols-[2.5rem_1fr_auto] items-center gap-3", children: [
    /* @__PURE__ */ jsx("div", { className: "h-10 w-10", children: showImage && /* @__PURE__ */ jsx(TokenImage, { token, size: tokenSize }) }),
    /* @__PURE__ */ jsxs("div", { className: "flex min-w-0 flex-col text-left", children: [
      /* @__PURE__ */ jsx(
        "span",
        {
          className: cn(
            text.headline,
            color.foreground,
            "overflow-hidden text-ellipsis whitespace-nowrap",
            classNames == null ? void 0 : classNames.tokenName
          ),
          children: (_a = token.name) == null ? void 0 : _a.trim()
        }
      ),
      /* @__PURE__ */ jsx(
        "span",
        {
          className: cn(
            text.label2,
            color.foregroundMuted,
            classNames == null ? void 0 : classNames.tokenValue
          ),
          children: `${formattedCryptoValue} ${token.symbol} ${subtitle ?? ""}`
        }
      )
    ] }),
    /* @__PURE__ */ jsx("div", { className: "text-right", children: !onActionPress && /* @__PURE__ */ jsx(
      "span",
      {
        className: cn(
          text.label2,
          color.foregroundMuted,
          "whitespace-nowrap",
          classNames == null ? void 0 : classNames.fiatValue
        ),
        children: formattedFiatValue
      }
    ) })
  ] });
}
function ActionButton({
  actionText,
  onActionPress,
  className
}) {
  return /* @__PURE__ */ jsx(
    "button",
    {
      type: "button",
      "data-testid": "ockTokenBalanceAction",
      "aria-label": actionText,
      onClick: onActionPress,
      className: cn(
        text.label2,
        color.primary,
        border.radius,
        "cursor-pointer p-0.5 font-bold",
        "border border-transparent hover:border-[--ock-line-primary]",
        "-translate-y-1/2 absolute top-1/2 right-2",
        className
      ),
      children: actionText
    }
  );
}
export {
  TokenBalance
};
