// @coinbase/onchainkit/dist/token/components/TokenSearch.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { useState, useCallback } from "react";
import { TextInput } from "../../internal/components/TextInput.js";
import { useTheme } from "../../internal/hooks/useTheme.js";
import { CloseSvg } from "../../internal/svg/closeSvg.js";
import { searchIconSvg } from "../../internal/svg/searchIconSvg.js";
import { cn, placeholder, color, pressable } from "../../styles/theme.js";
function TokenSearch({
  className,
  onChange,
  delayMs = 200
}) {
  const componentTheme = useTheme();
  const [value, setValue] = useState("");
  const handleClear = useCallback(() => {
    setValue("");
    onChange("");
  }, [onChange]);
  return /* @__PURE__ */ jsxs("div", { className: "relative flex items-center", children: [
    /* @__PURE__ */ jsx("div", { className: "-translate-y-1/2 absolute top-1/2 left-4", children: searchIconSvg }),
    /* @__PURE__ */ jsx(
      TextInput,
      {
        className: cn(
          componentTheme,
          pressable.alternate,
          color.foreground,
          placeholder.default,
          "w-full rounded-xl py-2 pr-5 pl-12 outline-none",
          className
        ),
        placeholder: "Search for a token",
        value,
        setValue,
        onChange,
        delayMs
      }
    ),
    value && /* @__PURE__ */ jsx(
      "button",
      {
        type: "button",
        "data-testid": "ockTextInput_Clear",
        className: "-translate-y-1/2 absolute top-1/2 right-4",
        onClick: handleClear,
        children: /* @__PURE__ */ jsx(CloseSvg, {})
      }
    )
  ] });
}
export {
  TokenSearch
};
