// @coinbase/onchainkit/dist/earn/components/VaultDetails.js
'use client';
import { jsx, jsxs } from "react/jsx-runtime";
import { getTruncatedAmount } from "../utils/getTruncatedAmount.js";
import { Popover } from "../../internal/components/Popover.js";
import { Skeleton } from "../../internal/components/Skeleton.js";
import { etherscanSvg } from "../../internal/svg/etherscanSvg.js";
import { infoSvg } from "../../internal/svg/infoSvg.js";
import { cn, color, background, border, text } from "../../styles/theme.js";
import "../../useOnchainKit.js";
import { useState, useRef } from "react";
import { TokenImage } from "../../token/components/TokenImage.js";
import "../../token/components/TokenRow.js";
import "../../internal/components/TextInput.js";
import "../../internal/svg/searchIconSvg.js";
import "react-dom";
import "../../token/components/TokenSelectButton.js";
import "viem";
import { useEarnContext } from "./EarnProvider.js";
function Row({ label, value }) {
  return /* @__PURE__ */ jsxs("div", { className: "flex items-center justify-between gap-8", children: [
    /* @__PURE__ */ jsx("div", { children: label }),
    /* @__PURE__ */ jsx("div", { className: "font-semibold", children: value })
  ] });
}
function VaultDetails() {
  const [isOpen, setIsOpen] = useState(false);
  const triggerRef = useRef(null);
  const anchorRef = useRef(null);
  const { vaultToken, vaultName, deposits, liquidity, vaultAddress } = useEarnContext();
  if (!vaultToken || !vaultName) {
    return /* @__PURE__ */ jsx(Skeleton, { className: "!rounded-full h-7 min-w-28" });
  }
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: anchorRef,
      className: cn(
        text.label1,
        color.foregroundMuted,
        background.alternate,
        "flex items-center justify-center gap-2 rounded-full p-1 px-3"
      ),
      "data-testid": "ock-vaultDetails",
      children: [
        /* @__PURE__ */ jsx(TokenImage, { token: vaultToken, size: 16 }),
        /* @__PURE__ */ jsx("span", { className: "max-w-24 truncate", title: vaultName, children: vaultName }),
        /* @__PURE__ */ jsx(
          "button",
          {
            ref: triggerRef,
            type: "button",
            "data-testid": "ock-vaultDetailsButton",
            className: cn(
              "size-3 [&_path]:fill-[var(--ock-icon-color-foreground-muted)] [&_path]:transition-colors [&_path]:ease-in-out [&_path]:hover:fill-[var(--ock-icon-color-foreground)]",
              isOpen && "[&_path]:fill-[var(--ock-icon-color-foreground)]"
            ),
            onClick: () => setIsOpen(!isOpen),
            children: infoSvg
          }
        ),
        /* @__PURE__ */ jsx(
          Popover,
          {
            isOpen,
            onClose: () => setIsOpen(false),
            position: "bottom",
            align: "start",
            trigger: triggerRef,
            anchor: anchorRef.current,
            offset: 4,
            children: /* @__PURE__ */ jsxs(
              "div",
              {
                className: cn(
                  background.default,
                  color.foreground,
                  border.defaultActive,
                  background.default,
                  "flex min-w-40 flex-col gap-3 rounded-lg border p-3 text-sm",
                  "fade-in animate-in duration-200"
                ),
                children: [
                  /* @__PURE__ */ jsx("div", { className: "font-semibold", children: vaultName }),
                  /* @__PURE__ */ jsxs("div", { className: "flex flex-col gap-2", children: [
                    /* @__PURE__ */ jsx(
                      Row,
                      {
                        label: "Token",
                        value: /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-1", children: [
                          /* @__PURE__ */ jsx(TokenImage, { token: vaultToken, size: 16 }),
                          /* @__PURE__ */ jsx("span", { className: "max-w-24 truncate", children: vaultToken.symbol })
                        ] })
                      }
                    ),
                    deposits ? /* @__PURE__ */ jsx(
                      Row,
                      {
                        label: "Total deposits",
                        value: `${getTruncatedAmount(deposits, 1, "compact")} ${vaultToken.symbol}`
                      }
                    ) : null,
                    liquidity ? /* @__PURE__ */ jsx(
                      Row,
                      {
                        label: "Liquidity",
                        value: `${getTruncatedAmount(liquidity, 1, "compact")} ${vaultToken.symbol}`
                      }
                    ) : null
                  ] }),
                  /* @__PURE__ */ jsxs(
                    "a",
                    {
                      href: `https://basescan.org/address/${vaultAddress}`,
                      target: "_blank",
                      rel: "noopener noreferrer",
                      className: cn(
                        color.primary,
                        "flex max-w-fit items-center gap-1 hover:opacity-80"
                      ),
                      "data-testid": "ock-vaultDetailsBaseScanLink",
                      children: [
                        /* @__PURE__ */ jsx("div", { children: "View on BaseScan" }),
                        /* @__PURE__ */ jsx("div", { className: "h-3 w-3", children: etherscanSvg })
                      ]
                    }
                  )
                ]
              }
            )
          }
        )
      ]
    }
  );
}
export {
  VaultDetails
};
