// @coinbase/onchainkit/dist/earn/components/YieldDetails.js
'use client';
import { jsx, jsxs } from "react/jsx-runtime";
import { useEarnContext } from "./EarnProvider.js";
import { getTruncatedAmount } from "../utils/getTruncatedAmount.js";
import { Popover } from "../../internal/components/Popover.js";
import { Skeleton } from "../../internal/components/Skeleton.js";
import { infoSvg } from "../../internal/svg/infoSvg.js";
import { formatPercent } from "../../internal/utils/formatPercent.js";
import { cn, background, color, text, border } from "../../styles/theme.js";
import { useState, useRef } from "react";
function YieldInfo() {
  const { rewards, nativeApy, vaultToken, vaultFee } = useEarnContext();
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        color.foregroundMuted,
        border.defaultActive,
        background.default,
        "fade-in flex min-w-52 animate-in flex-col gap-2 rounded-lg border p-3 text-sm duration-200"
      ),
      children: [
        nativeApy ? /* @__PURE__ */ jsxs(
          "div",
          {
            className: "flex items-center justify-between gap-1",
            "data-testid": "ock-earnNativeApy",
            children: [
              /* @__PURE__ */ jsx("div", { children: vaultToken == null ? void 0 : vaultToken.symbol }),
              /* @__PURE__ */ jsx("div", { className: "font-semibold", children: formatPercent(nativeApy) })
            ]
          }
        ) : null,
        rewards == null ? void 0 : rewards.map((reward) => /* @__PURE__ */ jsxs(
          "div",
          {
            className: "flex items-center justify-between gap-1",
            "data-testid": "ock-earnRewards",
            children: [
              /* @__PURE__ */ jsx("div", { children: reward.assetName }),
              /* @__PURE__ */ jsx("div", { className: "font-semibold", children: formatPercent(reward.apy) })
            ]
          },
          reward.asset
        )),
        vaultFee && nativeApy ? /* @__PURE__ */ jsxs(
          "div",
          {
            className: "flex items-center justify-between gap-1",
            "data-testid": "ock-earnPerformanceFee",
            children: [
              /* @__PURE__ */ jsxs("div", { children: [
                "Perf. Fee",
                " ",
                /* @__PURE__ */ jsxs("span", { className: "text-xs", children: [
                  "(",
                  formatPercent(vaultFee, 0),
                  ")"
                ] })
              ] }),
              /* @__PURE__ */ jsxs("div", { className: "font-semibold", children: [
                "-",
                formatPercent(vaultFee * nativeApy)
              ] })
            ]
          }
        ) : null
      ]
    }
  );
}
function YieldDetails() {
  const [isOpen, setIsOpen] = useState(false);
  const triggerRef = useRef(null);
  const anchorRef = useRef(null);
  const { apy } = useEarnContext();
  if (!apy) {
    return /* @__PURE__ */ jsx(Skeleton, { className: "!rounded-full h-7 min-w-28" });
  }
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref: anchorRef,
      className: cn(
        text.label1,
        color.foregroundMuted,
        background.alternate,
        "flex items-center justify-center gap-1 rounded-full p-1 px-3"
      ),
      "data-testid": "ock-yieldDetails",
      children: [
        `APY ${formatPercent(Number(getTruncatedAmount(apy.toString(), 4)))}`,
        /* @__PURE__ */ jsx(
          "button",
          {
            ref: triggerRef,
            type: "button",
            "data-testid": "ock-apyInfoButton",
            className: cn(
              "size-3 [&_path]:fill-[var(--ock-icon-color-foreground-muted)] [&_path]:transition-colors [&_path]:ease-in-out [&_path]:hover:fill-[var(--ock-icon-color-foreground)]",
              isOpen && "[&_path]:fill-[var(--ock-icon-color-foreground)]"
            ),
            onClick: () => setIsOpen(!isOpen),
            children: infoSvg
          }
        ),
        /* @__PURE__ */ jsx(
          Popover,
          {
            isOpen,
            onClose: () => setIsOpen(false),
            position: "bottom",
            align: "end",
            trigger: triggerRef,
            anchor: anchorRef.current,
            offset: 4,
            children: /* @__PURE__ */ jsx(YieldInfo, {})
          }
        )
      ]
    }
  );
}
export {
  YieldDetails
};
