// @coinbase/onchainkit/dist/identity/components/Avatar.js
'use client';
import { jsx, jsxs } from "react/jsx-runtime";
import { useIdentityContext } from "./IdentityProvider.js";
import { useAvatar } from "../hooks/useAvatar.js";
import { useName } from "../hooks/useName.js";
import { findComponent } from "../../internal/utils/findComponent.js";
import { useMemo, Children } from "react";
import { defaultAvatarSVG } from "../../internal/svg/defaultAvatarSVG.js";
import { defaultLoadingSVG } from "../../internal/svg/defaultLoadingSVG.js";
import { cn, border } from "../../styles/theme.js";
import { Badge } from "./Badge.js";
import { DisplayBadge } from "./DisplayBadge.js";
function Avatar({
  address = null,
  chain,
  className,
  defaultComponent,
  loadingComponent,
  children,
  ...props
}) {
  const { address: contextAddress, chain: contextChain } = useIdentityContext();
  const accountAddress = address ?? contextAddress;
  const accountChain = chain ?? contextChain;
  if (!accountAddress) {
    console.error(
      "Avatar: an Ethereum address must be provided to the Identity or Avatar component."
    );
    return null;
  }
  const { data: name, isLoading: isLoadingName } = useName({
    address: accountAddress,
    chain: accountChain
  });
  const { data: avatar, isLoading: isLoadingAvatar } = useAvatar(
    { ensName: name ?? "", chain: accountChain },
    { enabled: !!name }
  );
  const badge = useMemo(() => {
    return Children.toArray(children).find(findComponent(Badge));
  }, [children]);
  const defaultAvatar = defaultComponent || defaultAvatarSVG;
  const loadingAvatar = loadingComponent || defaultLoadingSVG;
  if (isLoadingName || isLoadingAvatar) {
    return /* @__PURE__ */ jsx("div", { className: cn("h-8 w-8 overflow-hidden rounded-full", className), children: loadingAvatar });
  }
  const displayAvatarImg = name && avatar;
  return /* @__PURE__ */ jsxs("div", { className: "relative", children: [
    /* @__PURE__ */ jsx(
      "div",
      {
        "data-testid": "ockAvatar_ImageContainer",
        className: cn("h-10 w-10 overflow-hidden rounded-full", className),
        children: displayAvatarImg ? /* @__PURE__ */ jsx(
          "img",
          {
            className: "min-h-full min-w-full object-cover",
            "data-testid": "ockAvatar_Image",
            loading: "lazy",
            width: "100%",
            height: "100%",
            decoding: "async",
            src: avatar,
            alt: name,
            ...props
          }
        ) : /* @__PURE__ */ jsx("div", { className: cn(border.default, "h-full w-full border"), children: defaultAvatar })
      }
    ),
    badge && /* @__PURE__ */ jsx(DisplayBadge, { address: accountAddress, children: /* @__PURE__ */ jsx(
      "div",
      {
        "data-testid": "ockAvatar_BadgeContainer",
        className: "-bottom-0.5 -right-0.5 absolute flex h-[15px] w-[15px] items-center justify-center rounded-full bg-transparent",
        children: /* @__PURE__ */ jsx("div", { className: "flex h-3 w-3 items-center justify-center", children: badge })
      }
    ) })
  ] });
}
export {
  Avatar
};
