// @coinbase/onchainkit/dist/identity/components/IdentityLayout.js
import { jsxs, jsx } from "react/jsx-runtime";
import { useTheme } from "../../internal/hooks/useTheme.js";
import { findComponent } from "../../internal/utils/findComponent.js";
import { cn, color, background } from "../../styles/theme.js";
import { useMemo, Children, cloneElement } from "react";
import { Address } from "./Address.js";
import { Avatar } from "./Avatar.js";
import { EthBalance } from "./EthBalance.js";
import { Name } from "./Name.js";
import { Socials } from "./Socials.js";
function IdentityLayout({
  children,
  className,
  hasCopyAddressOnClick
}) {
  const componentTheme = useTheme();
  const {
    avatar,
    name,
    address: addressComponent,
    ethBalance,
    socials
  } = useMemo(() => {
    const childrenArray = Children.toArray(children);
    const addressElement = childrenArray.find(findComponent(Address));
    return {
      avatar: childrenArray.find(findComponent(Avatar)),
      name: childrenArray.find(findComponent(Name)),
      address: addressElement ? cloneElement(addressElement, { hasCopyAddressOnClick }) : void 0,
      ethBalance: childrenArray.find(findComponent(EthBalance)),
      socials: childrenArray.find(findComponent(Socials))
    };
  }, [children, hasCopyAddressOnClick]);
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cn(
        componentTheme,
        background.default,
        "flex flex-col px-4 py-1",
        className
      ),
      "data-testid": "ockIdentityLayout_container",
      children: [
        /* @__PURE__ */ jsxs("div", { className: "flex items-center space-x-3", children: [
          /* @__PURE__ */ jsx("div", { className: "flex-shrink-0", children: avatar }),
          /* @__PURE__ */ jsxs("div", { className: "flex flex-col", children: [
            name,
            addressComponent && !ethBalance && addressComponent,
            !addressComponent && ethBalance && ethBalance,
            addressComponent && ethBalance && /* @__PURE__ */ jsxs("div", { className: "flex items-center gap-1", children: [
              addressComponent,
              /* @__PURE__ */ jsx("span", { className: color.foregroundMuted, children: "·" }),
              ethBalance
            ] })
          ] })
        ] }),
        socials
      ]
    }
  );
}
export {
  IdentityLayout
};
