// @coinbase/onchainkit/dist/nft/components/view/NFTMedia.js
import { jsx, jsxs } from "react/jsx-runtime";
import { useNFTLifecycleContext } from "../NFTLifecycleProvider.js";
import { useNFTContext } from "../NFTProvider.js";
import { MediaType, LifecycleType } from "../../types.js";
import { useMemo, useCallback } from "react";
import { cn } from "../../../styles/theme.js";
import { NFTAudio } from "./NFTAudio.js";
import { NFTImage } from "./NFTImage.js";
import { NFTVideo } from "./NFTVideo.js";
function NFTMedia({ className, square }) {
  const { mimeType } = useNFTContext();
  const { type, updateLifecycleStatus } = useNFTLifecycleContext();
  const mediaType = useMemo(() => {
    if (mimeType == null ? void 0 : mimeType.startsWith("video")) {
      return MediaType.Video;
    }
    if (mimeType == null ? void 0 : mimeType.startsWith("audio")) {
      return MediaType.Audio;
    }
    if (mimeType == null ? void 0 : mimeType.startsWith("image")) {
      return MediaType.Image;
    }
    return MediaType.Unknown;
  }, [mimeType]);
  const handleLoading = useCallback(
    (mediaUrl) => {
      updateLifecycleStatus({
        statusName: "mediaLoading",
        statusData: {
          mediaType,
          mediaUrl
        }
      });
    },
    [mediaType, updateLifecycleStatus]
  );
  const handleLoaded = useCallback(() => {
    updateLifecycleStatus({
      statusName: type === LifecycleType.MINT ? "mediaLoaded" : "success"
    });
  }, [type, updateLifecycleStatus]);
  const handleError = useCallback(
    (error) => {
      updateLifecycleStatus({
        statusName: "error",
        statusData: error
      });
    },
    [updateLifecycleStatus]
  );
  const media = useMemo(() => {
    switch (mediaType) {
      case MediaType.Video:
        return /* @__PURE__ */ jsx(
          NFTVideo,
          {
            square,
            onLoading: handleLoading,
            onLoaded: handleLoaded,
            onError: handleError
          }
        );
      case MediaType.Audio:
        return /* @__PURE__ */ jsxs("div", { className: "relative w-full", children: [
          /* @__PURE__ */ jsx(NFTImage, { square }),
          /* @__PURE__ */ jsx("div", { className: "absolute bottom-4 mx-auto w-full", children: /* @__PURE__ */ jsx(NFTAudio, {}) })
        ] });
      default:
        return /* @__PURE__ */ jsx(
          NFTImage,
          {
            square,
            onLoading: handleLoading,
            onLoaded: handleLoaded,
            onError: handleError
          }
        );
    }
  }, [handleError, handleLoaded, handleLoading, mediaType, square]);
  return /* @__PURE__ */ jsx("div", { className: cn("pb-2", className), children: media });
}
export {
  NFTMedia
};
