// @coinbase/onchainkit/dist/wallet/components/Wallet.js
'use client';
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { Draggable } from "../../internal/components/Draggable/Draggable.js";
import { useIsMounted } from "../../internal/hooks/useIsMounted.js";
import { useOutsideClick } from "../../internal/hooks/useOutsideClick.js";
import { useTheme } from "../../internal/hooks/useTheme.js";
import { findComponent } from "../../internal/utils/findComponent.js";
import { cn } from "../../styles/theme.js";
import { useRef, useMemo, Children, isValidElement } from "react";
import { getWalletDraggableProps } from "../utils/getWalletDraggableProps.js";
import { ConnectWallet } from "./ConnectWallet.js";
import { WalletAdvanced } from "./WalletAdvanced.js";
import { WalletDropdown } from "./WalletDropdown.js";
import { WalletProvider, useWalletContext } from "./WalletProvider.js";
const defaultWalletChildren = /* @__PURE__ */ jsxs(Fragment, { children: [
  /* @__PURE__ */ jsx(ConnectWallet, {}),
  /* @__PURE__ */ jsx(WalletDropdown, {})
] });
function Wallet({
  children,
  className,
  draggable,
  draggableStartingPosition,
  isSponsored
}) {
  const componentTheme = useTheme();
  const isMounted = useIsMounted();
  if (!isMounted) {
    return null;
  }
  return /* @__PURE__ */ jsx(WalletProvider, { isSponsored, children: /* @__PURE__ */ jsx(
    WalletContent,
    {
      className: cn(componentTheme, className),
      ...getWalletDraggableProps({ draggable, draggableStartingPosition }),
      children
    }
  ) });
}
function WalletContent({
  children,
  className,
  draggable,
  draggableStartingPosition
}) {
  const {
    isSubComponentOpen,
    isConnectModalOpen,
    handleClose,
    connectRef,
    breakpoint
  } = useWalletContext();
  const walletContainerRef = useRef(null);
  useOutsideClick(walletContainerRef, handleClose);
  const { dropdown, advanced } = useMemo(() => {
    const childrenArray = Children.toArray(children);
    return {
      dropdown: childrenArray.find(findComponent(WalletDropdown)),
      advanced: childrenArray.find(findComponent(WalletAdvanced))
    };
  }, [children]);
  const childrenToRender = useMemo(() => {
    return Children.map(children, (child) => {
      if (isValidElement(child) && child.type === WalletAdvanced && dropdown) {
        return null;
      }
      return child;
    });
  }, [dropdown, children]);
  if (dropdown && advanced) {
    console.error(
      "Defaulted to WalletDropdown. Wallet cannot have both WalletDropdown and WalletAdvanced as children."
    );
  }
  const disableDraggable = isConnectModalOpen || breakpoint === "sm" && isSubComponentOpen;
  if (draggable) {
    return /* @__PURE__ */ jsx(
      "div",
      {
        ref: walletContainerRef,
        className: cn("relative w-fit shrink-0", className),
        children: /* @__PURE__ */ jsx(
          Draggable,
          {
            startingPosition: draggableStartingPosition,
            disabled: disableDraggable,
            children: /* @__PURE__ */ jsx("div", { ref: connectRef, children: childrenToRender || defaultWalletChildren })
          }
        )
      }
    );
  }
  return /* @__PURE__ */ jsx(
    "div",
    {
      ref: walletContainerRef,
      className: cn("relative w-fit shrink-0", className),
      children: /* @__PURE__ */ jsx("div", { ref: connectRef, children: childrenToRender || defaultWalletChildren })
    }
  );
}
export {
  Wallet
};
