// @coinbase/onchainkit/dist/wallet/components/WalletAdvancedWalletActions.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { useAnalytics } from "../../core/analytics/hooks/useAnalytics.js";
import { WalletEvent, WalletOption } from "../../core/analytics/types.js";
import { PressableIcon } from "../../internal/components/PressableIcon.js";
import { baseScanSvg } from "../../internal/svg/baseScanSvg.js";
import { disconnectSvg } from "../../internal/svg/disconnectSvg.js";
import { qrIconSvg } from "../../internal/svg/qrIconSvg.js";
import { refreshSvg } from "../../internal/svg/refreshSvg.js";
import { cn } from "../../styles/theme.js";
import { useCallback } from "react";
import { useAccount, useDisconnect } from "wagmi";
import { useWalletContext } from "./WalletProvider.js";
import { usePortfolio } from "../hooks/usePortfolio.js";
import { RequestContext } from "../../core/network/constants.js";
function WalletAdvancedWalletActions({
  classNames
}) {
  const { address } = useAccount();
  const { handleClose, setActiveFeature, animations } = useWalletContext();
  const { disconnect, connectors } = useDisconnect();
  const { sendAnalytics } = useAnalytics();
  const { refetch: refetchPortfolioData } = usePortfolio(
    { address },
    RequestContext.Wallet
  );
  const handleAnalyticsOptionSelected = useCallback(
    (option) => {
      sendAnalytics(WalletEvent.OptionSelected, {
        option
      });
    },
    [sendAnalytics]
  );
  const handleAnalyticsDisconnect = useCallback(
    (walletProvider) => {
      sendAnalytics(WalletEvent.Disconnect, {
        component: "WalletAdvanced",
        walletProvider
      });
    },
    [sendAnalytics]
  );
  const handleTransactions = useCallback(() => {
    handleAnalyticsOptionSelected(WalletOption.Explorer);
    window.open(`https://basescan.org/address/${address}`, "_blank");
  }, [address, handleAnalyticsOptionSelected]);
  const handleDisconnect = useCallback(() => {
    var _a;
    const walletProvider = ((_a = connectors[0]) == null ? void 0 : _a.name) || "unknown";
    handleAnalyticsDisconnect(walletProvider);
    handleClose();
    for (const connector of connectors) {
      disconnect({ connector });
    }
  }, [disconnect, connectors, handleClose, handleAnalyticsDisconnect]);
  const handleQr = useCallback(() => {
    handleAnalyticsOptionSelected(WalletOption.QR);
    setActiveFeature("qr");
  }, [setActiveFeature, handleAnalyticsOptionSelected]);
  const handleRefreshPortfolioData = useCallback(async () => {
    handleAnalyticsOptionSelected(WalletOption.Refresh);
    await refetchPortfolioData();
  }, [refetchPortfolioData, handleAnalyticsOptionSelected]);
  return /* @__PURE__ */ jsxs(
    "div",
    {
      "data-testid": "ockWalletAdvanced_WalletActions",
      className: cn(
        "flex w-88 items-center justify-between px-4 py-3",
        animations.content,
        classNames == null ? void 0 : classNames.container
      ),
      children: [
        /* @__PURE__ */ jsxs("div", { className: "flex items-center", children: [
          /* @__PURE__ */ jsx(
            PressableIcon,
            {
              ariaLabel: "Open transaction history",
              onClick: handleTransactions,
              children: /* @__PURE__ */ jsx(
                "div",
                {
                  "data-testid": "ockWalletAdvanced_TransactionsButton",
                  className: cn("h-7 w-7 scale-110 p-2", classNames == null ? void 0 : classNames.baseScanIcon),
                  children: baseScanSvg
                }
              )
            }
          ),
          /* @__PURE__ */ jsx(PressableIcon, { ariaLabel: "Show QR code", onClick: handleQr, children: /* @__PURE__ */ jsx(
            "div",
            {
              "data-testid": "ockWalletAdvanced_QrButton",
              className: cn("h-7 w-7 scale-110", classNames == null ? void 0 : classNames.qrIcon),
              children: qrIconSvg
            }
          ) })
        ] }),
        /* @__PURE__ */ jsxs("div", { className: "flex items-center", children: [
          /* @__PURE__ */ jsx(PressableIcon, { ariaLabel: "Disconnect wallet", onClick: handleDisconnect, children: /* @__PURE__ */ jsx(
            "div",
            {
              "data-testid": "ockWalletAdvanced_DisconnectButton",
              className: cn("h-7 w-7 scale-110 p-2", classNames == null ? void 0 : classNames.disconnectIcon),
              children: disconnectSvg
            }
          ) }),
          /* @__PURE__ */ jsx(
            PressableIcon,
            {
              ariaLabel: "Refresh portfolio data",
              onClick: handleRefreshPortfolioData,
              children: /* @__PURE__ */ jsx(
                "div",
                {
                  "data-testid": "ockWalletAdvanced_RefreshButton",
                  className: cn("h-7 w-7 scale-110 p-2", classNames == null ? void 0 : classNames.refreshIcon),
                  children: refreshSvg
                }
              )
            }
          )
        ] })
      ]
    }
  );
}
export {
  WalletAdvancedWalletActions
};
