// @coinbase/onchainkit/dist/wallet/components/WalletAdvancedQrReceive.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { CopyButton } from "../../internal/components/CopyButton.js";
import { PressableIcon } from "../../internal/components/PressableIcon.js";
import { QrCodeSvg } from "../../internal/components/QrCode/QrCodeSvg.js";
import { backArrowSvg } from "../../internal/svg/backArrowSvg.js";
import { copySvg } from "../../internal/svg/copySvg.js";
import { zIndex } from "../../styles/constants.js";
import { cn, border, pressable, color, text } from "../../styles/theme.js";
import { useState, useCallback } from "react";
import { useWalletContext } from "./WalletProvider.js";
function WalletAdvancedQrReceive({
  classNames
}) {
  const {
    address,
    setActiveFeature,
    isActiveFeatureClosing,
    setIsActiveFeatureClosing
  } = useWalletContext();
  const [copyText, setCopyText] = useState("Copy");
  const [copyButtonText, setCopyButtonText] = useState("Copy address");
  const handleCloseQr = useCallback(() => {
    setIsActiveFeatureClosing(true);
  }, [setIsActiveFeatureClosing]);
  const handleAnimationEnd = useCallback(() => {
    if (isActiveFeatureClosing) {
      setActiveFeature(null);
      setIsActiveFeatureClosing(false);
    }
  }, [isActiveFeatureClosing, setActiveFeature, setIsActiveFeatureClosing]);
  const resetAffordanceText = useCallback(() => {
    setTimeout(() => {
      setCopyText("Copy");
      setCopyButtonText("Copy address");
    }, 2e3);
  }, []);
  const handleCopyButtonSuccess = useCallback(() => {
    setCopyButtonText("Address copied");
    resetAffordanceText();
  }, [resetAffordanceText]);
  const handleCopyButtonError = useCallback(() => {
    setCopyButtonText("Failed to copy address");
    resetAffordanceText();
  }, [resetAffordanceText]);
  const handleCopyIconSuccess = useCallback(() => {
    setCopyText("Copied");
    resetAffordanceText();
  }, [resetAffordanceText]);
  const handleCopyIconError = useCallback(() => {
    setCopyText("Failed to copy");
    resetAffordanceText();
  }, [resetAffordanceText]);
  return /* @__PURE__ */ jsxs(
    "div",
    {
      "data-testid": "ockWalletAdvancedQrReceive",
      className: cn(
        border.radius,
        color.foreground,
        text.headline,
        "flex flex-col items-center justify-between",
        "h-120 w-88 px-4 pt-3 pb-4",
        isActiveFeatureClosing ? "fade-out slide-out-to-left-5 animate-out fill-mode-forwards ease-in-out" : "fade-in slide-in-from-left-5 linear animate-in duration-150",
        classNames == null ? void 0 : classNames.container
      ),
      onAnimationEnd: handleAnimationEnd,
      children: [
        /* @__PURE__ */ jsxs(
          "div",
          {
            className: cn(
              "flex h-[34px] w-full flex-row items-center justify-between",
              classNames == null ? void 0 : classNames.header
            ),
            children: [
              /* @__PURE__ */ jsx(PressableIcon, { ariaLabel: "Back button", onClick: handleCloseQr, children: /* @__PURE__ */ jsx("div", { className: "p-2", children: backArrowSvg }) }),
              /* @__PURE__ */ jsx("span", { children: "Scan to receive" }),
              /* @__PURE__ */ jsxs("div", { className: "group relative", children: [
                /* @__PURE__ */ jsx(
                  CopyButton,
                  {
                    label: copySvg,
                    copyValue: address ?? "",
                    onSuccess: handleCopyIconSuccess,
                    onError: handleCopyIconError,
                    className: cn(
                      pressable.default,
                      border.radiusInner,
                      border.default,
                      "flex items-center justify-center p-2"
                    ),
                    "aria-label": "Copy your address by clicking the icon"
                  }
                ),
                /* @__PURE__ */ jsx(
                  CopyButton,
                  {
                    label: copyText,
                    copyValue: address ?? "",
                    onSuccess: handleCopyIconSuccess,
                    onError: handleCopyIconError,
                    className: cn(
                      pressable.alternate,
                      text.legal,
                      color.foreground,
                      border.default,
                      border.radius,
                      zIndex.dropdown,
                      "absolute top-full right-0 mt-0.5 px-1.5 py-0.5 opacity-0 transition-opacity group-hover:opacity-100"
                    ),
                    "aria-label": "Copy your address by clicking the tooltip"
                  }
                )
              ] })
            ]
          }
        ),
        /* @__PURE__ */ jsx(QrCodeSvg, { value: address }),
        /* @__PURE__ */ jsx(
          CopyButton,
          {
            copyValue: address ?? "",
            label: copyButtonText,
            className: cn(
              border.radius,
              pressable.alternate,
              "w-full p-3",
              classNames == null ? void 0 : classNames.copyButton
            ),
            onSuccess: handleCopyButtonSuccess,
            onError: handleCopyButtonError,
            "aria-label": "Copy your address by clicking the button"
          }
        )
      ]
    }
  );
}
export {
  WalletAdvancedQrReceive
};
