// @coinbase/onchainkit/dist/wallet/components/WalletDropdown.js
'use client';
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { Address } from "../../identity/components/Address.js";
import { Avatar } from "../../identity/components/Avatar.js";
import "../../core/network/attestations.js";
import "../../identity/utils/easSupportedChains.js";
import "@tanstack/react-query";
import "../../internal/svg/badgeSvg.js";
import { cn, color } from "../../styles/theme.js";
import "react";
import "../../useOnchainKit.js";
import "../../identity/components/IdentityProvider.js";
import { EthBalance } from "../../identity/components/EthBalance.js";
import { Identity } from "../../identity/components/Identity.js";
import { Name } from "../../identity/components/Name.js";
import "viem/chains";
import "viem";
import "../../identity/constants.js";
import "../../identity/utils/getAddress.js";
import "viem/ens";
import "../../internal/svg/githubSvg.js";
import "../../internal/svg/twitterSvg.js";
import "../../internal/svg/warpcastSvg.js";
import "../../internal/svg/websiteSvg.js";
import "../../identity/utils/getAddresses.js";
import { WalletDropdownContent } from "./WalletDropdownContent.js";
import { WalletDropdownDisconnect } from "./WalletDropdownDisconnect.js";
import { WalletDropdownLink } from "./WalletDropdownLink.js";
import { useWalletContext } from "./WalletProvider.js";
const defaultWalletDropdownChildren = /* @__PURE__ */ jsxs(Fragment, { children: [
  /* @__PURE__ */ jsxs(Identity, { className: "px-4 pt-3 pb-2", children: [
    /* @__PURE__ */ jsx(Avatar, {}),
    /* @__PURE__ */ jsx(Name, {}),
    /* @__PURE__ */ jsx(Address, { className: color.foregroundMuted }),
    /* @__PURE__ */ jsx(EthBalance, {})
  ] }, "wallet-dd-identity"),
  /* @__PURE__ */ jsx(
    WalletDropdownLink,
    {
      icon: "wallet",
      href: "https://keys.coinbase.com",
      target: "_blank",
      children: "Wallet"
    },
    "wallet-dd-link"
  ),
  /* @__PURE__ */ jsx(WalletDropdownDisconnect, {}, "wallet-dd-disconnect")
] });
function WalletDropdown({
  children,
  className,
  classNames,
  swappableTokens
}) {
  const {
    address,
    breakpoint,
    isSubComponentOpen,
    showSubComponentAbove,
    alignSubComponentRight
  } = useWalletContext();
  if (!address) {
    return null;
  }
  if (!breakpoint) {
    return null;
  }
  if (!isSubComponentOpen) {
    return null;
  }
  return /* @__PURE__ */ jsx(
    "div",
    {
      "data-testid": "ockWalletDropdown",
      className: cn(
        "absolute",
        showSubComponentAbove ? "bottom-full" : "top-full",
        alignSubComponentRight ? "right-0" : "left-0",
        className
      ),
      children: /* @__PURE__ */ jsx(
        WalletDropdownContent,
        {
          classNames,
          swappableTokens,
          children: children || defaultWalletDropdownChildren
        }
      )
    }
  );
}
export {
  WalletDropdown
};
