// @coinbase/onchainkit/dist/wallet/components/WalletModal.js
'use client';
import { jsx, jsxs } from "react/jsx-runtime";
import { Dialog } from "../../internal/components/Dialog.js";
import { CloseSvg } from "../../internal/svg/closeSvg.js";
import { coinbaseWalletSvg } from "../../internal/svg/coinbaseWalletSvg.js";
import { defaultAvatarSVG } from "../../internal/svg/defaultAvatarSVG.js";
import { frameWalletSvg } from "../../internal/svg/frameWalletSvg.js";
import { metamaskSvg } from "../../internal/svg/metamaskSvg.js";
import { phantomSvg } from "../../internal/svg/phantomSvg.js";
import { rabbySvg } from "../../internal/svg/rabbySvg.js";
import { trustWalletSvg } from "../../internal/svg/trustWalletSvg.js";
import { cn, border, pressable, color, text, background } from "../../styles/theme.js";
import { useOnchainKit } from "../../useOnchainKit.js";
import { useCallback } from "react";
import { useConnect } from "wagmi";
import { baseAccount, coinbaseWallet, metaMask, injected } from "wagmi/connectors";
import { checkWalletAndRedirect } from "../utils/checkWalletAndRedirect.js";
import { BaseAccountSvg } from "../../internal/svg/baseAccountSvg.js";
function WalletModal({
  className,
  isOpen,
  onClose,
  onError
}) {
  var _a, _b, _c, _d, _e, _f;
  const { connect } = useConnect();
  const { config } = useOnchainKit();
  const appLogo = ((_a = config == null ? void 0 : config.appearance) == null ? void 0 : _a.logo) ?? void 0;
  const appName = ((_b = config == null ? void 0 : config.appearance) == null ? void 0 : _b.name) ?? void 0;
  const privacyPolicyUrl = ((_c = config == null ? void 0 : config.wallet) == null ? void 0 : _c.privacyUrl) ?? void 0;
  const termsOfServiceUrl = ((_d = config == null ? void 0 : config.wallet) == null ? void 0 : _d.termsUrl) ?? void 0;
  const supportedWallets = ((_e = config == null ? void 0 : config.wallet) == null ? void 0 : _e.supportedWallets) ?? {
    rabby: false,
    trust: false,
    frame: false
  };
  const isSignUpEnabled = ((_f = config == null ? void 0 : config.wallet) == null ? void 0 : _f.signUpEnabled) ?? true;
  const handleBaseAccountConnection = useCallback(() => {
    try {
      connect({
        connector: baseAccount({
          appName,
          appLogoUrl: appLogo
        })
      });
      onClose();
    } catch (error) {
      console.error("Base Account connection error:", error);
      if (onError) {
        onError(
          error instanceof Error ? error : new Error("Failed to connect wallet")
        );
      }
    }
  }, [appName, appLogo, connect, onClose, onError]);
  const handleCoinbaseWalletConnection = useCallback(() => {
    try {
      const cbConnector = coinbaseWallet({
        preference: "all",
        appName,
        appLogoUrl: appLogo
      });
      connect({ connector: cbConnector });
      onClose();
    } catch (error) {
      console.error("Coinbase Wallet connection error:", error);
      if (onError) {
        onError(
          error instanceof Error ? error : new Error("Failed to connect wallet")
        );
      }
    }
  }, [appName, appLogo, connect, onClose, onError]);
  const handleMetaMaskConnection = useCallback(() => {
    try {
      const metamaskConnector = metaMask({
        dappMetadata: {
          name: appName || "OnchainKit App",
          url: window.location.origin,
          iconUrl: appLogo
        }
      });
      connect({ connector: metamaskConnector });
      onClose();
    } catch (error) {
      console.error("MetaMask connection error:", error);
      onError == null ? void 0 : onError(
        error instanceof Error ? error : new Error("Failed to connect wallet")
      );
    }
  }, [connect, onClose, onError, appName, appLogo]);
  const handlePhantomConnection = useCallback(() => {
    try {
      if (!checkWalletAndRedirect("phantom")) {
        onClose();
        return;
      }
      const phantomConnector = injected({
        target: "phantom"
      });
      connect({ connector: phantomConnector });
      onClose();
    } catch (error) {
      console.error("Phantom connection error:", error);
      onError == null ? void 0 : onError(
        error instanceof Error ? error : new Error("Failed to connect wallet")
      );
    }
  }, [connect, onClose, onError]);
  const handleRabbyConnection = useCallback(() => {
    try {
      if (!checkWalletAndRedirect("rabby")) {
        onClose();
        return;
      }
      const rabbyConnector = injected({
        target: "rabby"
      });
      connect({ connector: rabbyConnector });
      onClose();
    } catch (error) {
      console.error("Rabby connection error:", error);
      onError == null ? void 0 : onError(
        error instanceof Error ? error : new Error("Failed to connect wallet")
      );
    }
  }, [connect, onClose, onError]);
  const handleTrustWalletConnection = useCallback(() => {
    try {
      if (!checkWalletAndRedirect("trust")) {
        onClose();
        return;
      }
      const trustConnector = injected({
        target: "trust"
      });
      connect({ connector: trustConnector });
      onClose();
    } catch (error) {
      console.error("Trust Wallet connection error:", error);
      onError == null ? void 0 : onError(
        error instanceof Error ? error : new Error("Failed to connect wallet")
      );
      onClose();
    }
  }, [connect, onClose, onError]);
  const handleFrameWalletConnection = useCallback(() => {
    var _a2;
    try {
      if (!((_a2 = window.ethereum) == null ? void 0 : _a2.isFrame)) {
        window.open("https://frame.sh/download", "_blank");
        onClose();
        return;
      }
      const frameConnector = injected();
      connect({ connector: frameConnector });
      onClose();
    } catch (error) {
      console.error("Frame Wallet connection error:", error);
      onError == null ? void 0 : onError(
        error instanceof Error ? error : new Error("Failed to connect wallet")
      );
      onClose();
    }
  }, [connect, onClose, onError]);
  const availableWallets = [
    {
      id: "base-account",
      name: "Base",
      icon: /* @__PURE__ */ jsx(BaseAccountSvg, {}),
      connector: handleBaseAccountConnection,
      enabled: true
    },
    {
      id: "coinbase",
      name: "Coinbase Wallet",
      icon: coinbaseWalletSvg,
      connector: handleCoinbaseWalletConnection,
      enabled: true
    },
    {
      id: "metamask",
      name: "MetaMask",
      icon: metamaskSvg,
      connector: handleMetaMaskConnection,
      enabled: true
    },
    {
      id: "phantom",
      name: "Phantom",
      icon: phantomSvg,
      connector: handlePhantomConnection,
      enabled: true
    },
    {
      id: "rabby",
      name: "Rabby",
      icon: rabbySvg,
      connector: handleRabbyConnection,
      enabled: supportedWallets.rabby === true
    },
    {
      id: "trust",
      name: "Trust Wallet",
      icon: trustWalletSvg,
      connector: handleTrustWalletConnection,
      enabled: supportedWallets.trust === true
    },
    {
      id: "frame",
      name: "Frame",
      icon: frameWalletSvg,
      connector: handleFrameWalletConnection,
      enabled: supportedWallets.frame === true
    }
  ].filter((wallet) => wallet.enabled);
  return /* @__PURE__ */ jsx(Dialog, { isOpen, onClose, "aria-label": "Connect Wallet", children: /* @__PURE__ */ jsxs(
    "div",
    {
      "data-testid": "ockModalOverlay",
      className: cn(
        border.lineDefault,
        border.radius,
        background.default,
        "w-[22rem] p-6 pb-4",
        "relative flex flex-col items-center gap-4",
        className
      ),
      children: [
        /* @__PURE__ */ jsx(
          "button",
          {
            type: "button",
            onClick: onClose,
            className: cn(
              pressable.default,
              border.radius,
              border.default,
              "absolute top-4 right-4",
              "flex items-center justify-center p-1",
              "transition-colors duration-200"
            ),
            "aria-label": "Close modal",
            children: /* @__PURE__ */ jsx("div", { className: cn("flex h-4 w-4 items-center justify-center"), children: /* @__PURE__ */ jsx(CloseSvg, {}) })
          }
        ),
        (appLogo || appName) && /* @__PURE__ */ jsxs("div", { className: "flex w-full flex-col items-center gap-2 py-3", children: [
          appLogo && /* @__PURE__ */ jsx("div", { className: cn(border.radius, "h-14 w-14 overflow-hidden"), children: /* @__PURE__ */ jsx(
            "img",
            {
              src: appLogo,
              alt: `${appName || "App"} icon`,
              className: "h-full w-full object-cover"
            }
          ) }),
          appName && /* @__PURE__ */ jsx(
            "h2",
            {
              className: cn(text.headline, color.foreground, "text-center"),
              children: appName
            }
          )
        ] }),
        /* @__PURE__ */ jsxs("div", { className: "flex w-full flex-col gap-3", children: [
          isSignUpEnabled && /* @__PURE__ */ jsxs(
            "button",
            {
              type: "button",
              onClick: handleCoinbaseWalletConnection,
              className: cn(
                border.radius,
                text.body,
                pressable.alternate,
                color.foreground,
                "flex items-center justify-between px-4 py-3 text-left"
              ),
              children: [
                "Sign up",
                /* @__PURE__ */ jsx("div", { className: "h-4 w-4", children: defaultAvatarSVG })
              ]
            }
          ),
          /* @__PURE__ */ jsxs("div", { className: "relative", children: [
            isSignUpEnabled && /* @__PURE__ */ jsx("div", { className: "absolute inset-0 flex items-center", children: /* @__PURE__ */ jsx(
              "div",
              {
                className: cn(border.lineDefault, "w-full border-[0.5px]")
              }
            ) }),
            /* @__PURE__ */ jsx("div", { className: "relative flex justify-center", children: /* @__PURE__ */ jsx(
              "span",
              {
                className: cn(
                  background.default,
                  color.foregroundMuted,
                  text.legal,
                  "px-2"
                ),
                children: isSignUpEnabled ? "or continue with an existing wallet" : "Connect your wallet"
              }
            ) })
          ] }),
          availableWallets.map((wallet) => /* @__PURE__ */ jsxs(
            "button",
            {
              type: "button",
              onClick: wallet.connector,
              className: cn(
                border.radius,
                background.default,
                text.body,
                pressable.alternate,
                color.foreground,
                "flex items-center justify-between px-4 py-3 text-left"
              ),
              children: [
                wallet.name,
                /* @__PURE__ */ jsx("div", { className: "-mr-0.5 flex h-4 w-4 items-center justify-center", children: wallet.icon })
              ]
            },
            wallet.id
          ))
        ] }),
        /* @__PURE__ */ jsxs(
          "div",
          {
            className: cn(
              color.foregroundMuted,
              text.legal,
              "flex flex-col items-center justify-center gap-1 px-4",
              "mt-4 text-center"
            ),
            children: [
              /* @__PURE__ */ jsx("span", { className: "font-normal text-[10px] leading-[13px]", children: "By connecting a wallet, you agree to our" }),
              /* @__PURE__ */ jsxs("span", { className: "font-normal text-[10px] leading-[13px]", children: [
                termsOfServiceUrl && /* @__PURE__ */ jsx(
                  "a",
                  {
                    href: termsOfServiceUrl,
                    className: cn(color.primary, "hover:underline"),
                    target: "_blank",
                    rel: "noopener noreferrer",
                    tabIndex: 0,
                    children: "Terms of Service"
                  }
                ),
                " ",
                termsOfServiceUrl && privacyPolicyUrl && "and",
                " ",
                privacyPolicyUrl && /* @__PURE__ */ jsx(
                  "a",
                  {
                    href: privacyPolicyUrl,
                    className: cn(color.primary, "hover:underline"),
                    target: "_blank",
                    rel: "noopener noreferrer",
                    tabIndex: 0,
                    children: "Privacy Policy"
                  }
                ),
                "."
              ] })
            ]
          }
        )
      ]
    }
  ) });
}
export {
  WalletModal
};
