// @coinbase/onchainkit/dist/wallet/components/WalletIsland.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import "react";
import "clsx";
import "tailwind-merge";
import "../../identity/components/IdentityProvider.js";
import { Avatar } from "../../identity/components/Avatar.js";
import "../../core/network/attestations.js";
import "../../identity/utils/easSupportedChains.js";
import "@tanstack/react-query";
import "../../internal/svg/badgeSvg.js";
import "../../useOnchainKit.js";
import "viem";
import "wagmi";
import "viem/chains";
import "../../identity/constants.js";
import "../../identity/utils/getAddress.js";
import "viem/ens";
import "../../internal/svg/githubSvg.js";
import "../../internal/svg/twitterSvg.js";
import "../../internal/svg/warpcastSvg.js";
import "../../internal/svg/websiteSvg.js";
import "../../identity/utils/getAddresses.js";
import { portfolioSvg } from "../../internal/svg/portfolioSvg.js";
import { getDefaultDraggableStartingPosition } from "../utils/getDefaultDraggableStartingPosition.js";
import { ConnectWallet } from "./ConnectWallet.js";
import { Wallet } from "./Wallet.js";
import { WalletAdvancedAddressDetails } from "./WalletAdvancedAddressDetails.js";
import { WalletAdvancedTokenHoldings } from "./WalletAdvancedTokenHoldings.js";
import { WalletAdvancedTransactionActions } from "./WalletAdvancedTransactionActions.js";
import { WalletAdvancedWalletActions } from "./WalletAdvancedWalletActions.js";
import { WalletDropdown } from "./WalletDropdown.js";
function WalletIsland({
  startingPosition = getDefaultDraggableStartingPosition(),
  isSponsored
}) {
  return /* @__PURE__ */ jsxs(
    Wallet,
    {
      draggable: true,
      draggableStartingPosition: startingPosition,
      isSponsored,
      children: [
        /* @__PURE__ */ jsx(
          ConnectWallet,
          {
            className: "!rounded-full m-0 flex h-14 w-14 min-w-14 flex-col items-center justify-center p-0",
            disconnectedLabel: /* @__PURE__ */ jsx("div", { className: "h-5 w-5", children: portfolioSvg }),
            children: /* @__PURE__ */ jsx(Avatar, { className: "pointer-events-none h-14 w-14" })
          }
        ),
        /* @__PURE__ */ jsxs(WalletDropdown, { children: [
          /* @__PURE__ */ jsx(WalletAdvancedWalletActions, {}),
          /* @__PURE__ */ jsx(WalletAdvancedAddressDetails, {}),
          /* @__PURE__ */ jsx(WalletAdvancedTransactionActions, {}),
          /* @__PURE__ */ jsx(WalletAdvancedTokenHoldings, {})
        ] })
      ]
    }
  );
}
export {
  WalletIsland
};
