// @coinbase/onchainkit/dist/appchain/bridge/components/AppchainBridgeSuccess.js
'use client';
import { jsxs, jsx } from "react/jsx-runtime";
import { SuccessSvg } from "../../../internal/svg/fullWidthSuccessSvg.js";
import { color, cn, text, border, pressable } from "../../../styles/theme.js";
import { useAppchainBridgeContext } from "./AppchainBridgeProvider.js";
const AppchainBridgeSuccess = ({
  title = "Success!",
  primaryButtonLabel = "View Transaction",
  secondaryButtonLabel = "Back to bridge"
}) => {
  const { handleOpenExplorer, handleResetState } = useAppchainBridgeContext();
  return /* @__PURE__ */ jsxs("div", { className: "flex h-full w-full flex-col justify-between", children: [
    /* @__PURE__ */ jsx("div", { children: /* @__PURE__ */ jsx("div", { className: "flex items-center" }) }),
    /* @__PURE__ */ jsx("div", { className: "mt-16 px-4", children: /* @__PURE__ */ jsxs("div", { className: "flex flex-col items-center gap-16", children: [
      /* @__PURE__ */ jsxs("div", { className: "mb-6 flex flex-col items-center gap-4", children: [
        /* @__PURE__ */ jsx("div", { className: "flex justify-center", children: /* @__PURE__ */ jsx("div", { className: "h-12 w-12", children: /* @__PURE__ */ jsx(SuccessSvg, { fill: "var(--ock-bg-primary)" }) }) }),
        /* @__PURE__ */ jsx("div", { className: "ock-text-foreground flex-1 text-center font-medium text-sm", children: title })
      ] }),
      /* @__PURE__ */ jsx("div", { className: "flex w-full flex-col items-center gap-2", children: [
        {
          label: primaryButtonLabel,
          action: handleOpenExplorer,
          variant: "primary",
          textColor: color.inverse
        },
        {
          label: secondaryButtonLabel,
          action: handleResetState,
          variant: "secondary",
          textColor: color.foreground
        }
      ].map(({ label, action, variant, textColor }) => /* @__PURE__ */ jsx(
        "button",
        {
          className: cn(
            pressable[variant],
            border.radius,
            "w-full rounded-xl",
            "px-4 py-3 text-base text-white leading-6",
            text.label1
          ),
          type: "button",
          onClick: action,
          children: /* @__PURE__ */ jsx(
            "div",
            {
              className: cn(
                text.headline,
                textColor,
                "flex justify-center"
              ),
              children: label
            }
          )
        },
        label
      )) })
    ] }) })
  ] });
};
export {
  AppchainBridgeSuccess
};
