// @coinbase/onchainkit/dist/fund/components/FundCardPaymentMethodSelectRow.js
import { jsx, jsxs } from "react/jsx-runtime";
import { memo, useCallback } from "react";
import { useAnalytics } from "../../core/analytics/hooks/useAnalytics.js";
import { FundEvent } from "../../core/analytics/types.js";
import { cn, pressable, text, color, background, border } from "../../styles/theme.js";
import { FundCardPaymentMethodImage } from "./FundCardPaymentMethodImage.js";
const FundCardPaymentMethodSelectRow = memo(
  ({
    paymentMethod,
    onClick,
    hideImage,
    hideDescription,
    disabled,
    disabledReason,
    testId
  }) => {
    const { sendAnalytics } = useAnalytics();
    const handleOnClick = useCallback(() => {
      if (!disabled) {
        onClick == null ? void 0 : onClick(paymentMethod);
        sendAnalytics(FundEvent.FundOptionSelected, {
          option: paymentMethod.id
        });
      }
    }, [disabled, onClick, paymentMethod, sendAnalytics]);
    return /* @__PURE__ */ jsx(
      "button",
      {
        "data-testid": testId,
        type: "button",
        className: cn(
          pressable.default,
          border.radius,
          background.default,
          "flex w-full items-center justify-between px-4 py-2",
          {
            [pressable.disabled]: disabled
          }
        ),
        onClick: handleOnClick,
        disabled,
        title: disabledReason,
        children: /* @__PURE__ */ jsxs("span", { className: "flex items-center gap-3", children: [
          !hideImage && /* @__PURE__ */ jsx(
            FundCardPaymentMethodImage,
            {
              paymentMethod,
              className: cn("h-4 w-4", {
                [pressable.disabled]: disabled
              })
            }
          ),
          /* @__PURE__ */ jsxs("span", { className: "flex flex-col items-start", children: [
            /* @__PURE__ */ jsx("span", { className: cn(text.headline), children: paymentMethod.name }),
            !hideDescription && /* @__PURE__ */ jsx(
              "span",
              {
                className: cn(
                  text.label2,
                  color.foregroundMuted,
                  "font-normal"
                ),
                children: disabledReason || paymentMethod.description
              }
            )
          ] })
        ] })
      }
    );
  }
);
FundCardPaymentMethodSelectRow.displayName = "FundCardPaymentMethodSelectRow";
export {
  FundCardPaymentMethodSelectRow
};
