// @coinbase/onchainkit/dist/fund/components/FundButton.js
'use client';
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useCallback, useMemo } from "react";
import { useTheme } from "../../internal/hooks/useTheme.js";
import { background, pressable, cn, color, border, text } from "../../styles/theme.js";
import { usePopupMonitor } from "../../buy/hooks/usePopupMonitor.js";
import { ErrorSvg } from "../../internal/svg/errorSvg.js";
import { openPopup } from "../../internal/utils/openPopup.js";
import { useAccount } from "wagmi";
import { useAnalytics } from "../../core/analytics/hooks/useAnalytics.js";
import { FundEvent } from "../../core/analytics/types.js";
import { Spinner } from "../../internal/components/Spinner.js";
import { AddSvg } from "../../internal/svg/addSvg.js";
import { SuccessSvg } from "../../internal/svg/successSvg.js";
import { ConnectWallet } from "../../wallet/components/ConnectWallet.js";
import { useGetFundingUrl } from "../hooks/useGetFundingUrl.js";
import { getFundingPopupSize } from "../utils/getFundingPopupSize.js";
function FundButton({
  className,
  disabled = false,
  fundingUrl,
  hideIcon = false,
  hideText = false,
  openIn = "popup",
  popupSize = "md",
  rel,
  target,
  text: buttonText = "Fund",
  successText: buttonSuccessText = "Success",
  errorText: buttonErrorText = "Something went wrong",
  state: buttonState = "default",
  fiatCurrency = "USD",
  onPopupClose,
  onClick
}) {
  const componentTheme = useTheme();
  const fallbackFundingUrl = useGetFundingUrl({
    fiatCurrency,
    originComponentName: "FundButton"
  });
  const { address } = useAccount();
  const fundingUrlToRender = fundingUrl ?? fallbackFundingUrl;
  const isDisabled = disabled || !fundingUrlToRender;
  const shouldShowConnectWallet = !address;
  const { startPopupMonitor } = usePopupMonitor(onPopupClose);
  const { sendAnalytics } = useAnalytics();
  const handleAnalyticsInitiated = useCallback(() => {
    sendAnalytics(FundEvent.FundInitiated, {
      currency: fiatCurrency
    });
  }, [sendAnalytics, fiatCurrency]);
  const handleAnalyticsFailure = useCallback(
    (error) => {
      sendAnalytics(FundEvent.FundFailure, {
        error,
        metadata: { currency: fiatCurrency }
      });
    },
    [sendAnalytics, fiatCurrency]
  );
  const handleClick = useCallback(
    (e) => {
      e.preventDefault();
      if (fundingUrlToRender) {
        handleAnalyticsInitiated();
        onClick == null ? void 0 : onClick();
        const { height, width } = getFundingPopupSize(
          popupSize,
          fundingUrlToRender
        );
        const popupWindow = openPopup({
          url: fundingUrlToRender,
          height,
          width,
          target
        });
        if (popupWindow) {
          startPopupMonitor(popupWindow);
        } else {
          handleAnalyticsFailure("Failed to open funding popup");
        }
      }
    },
    [
      fundingUrlToRender,
      popupSize,
      target,
      onClick,
      startPopupMonitor,
      handleAnalyticsInitiated,
      handleAnalyticsFailure
    ]
  );
  const buttonColorClass = useMemo(() => {
    if (buttonState === "error") {
      return background.error;
    }
    return pressable.primary;
  }, [buttonState]);
  const classNames = cn(
    componentTheme,
    buttonColorClass,
    "px-4 py-3 inline-flex items-center justify-center space-x-2",
    {
      [pressable.disabled]: isDisabled
    },
    text.headline,
    border.radius,
    color.inverse,
    className
  );
  const buttonIcon = useMemo(() => {
    if (hideIcon) {
      return null;
    }
    switch (buttonState) {
      case "loading":
        return "";
      case "success":
        return /* @__PURE__ */ jsx(SuccessSvg, { fill: "#F9FAFB" });
      case "error":
        return /* @__PURE__ */ jsx(ErrorSvg, { fill: "#F9FAFB" });
      default:
        return /* @__PURE__ */ jsx(AddSvg, {});
    }
  }, [buttonState, hideIcon]);
  const buttonTextContent = useMemo(() => {
    switch (buttonState) {
      case "loading":
        return "";
      case "success":
        return buttonSuccessText;
      case "error":
        return buttonErrorText;
      default:
        return buttonText;
    }
  }, [buttonState, buttonSuccessText, buttonErrorText, buttonText]);
  const buttonContent = useMemo(() => {
    if (buttonState === "loading") {
      return /* @__PURE__ */ jsx(Spinner, {});
    }
    return /* @__PURE__ */ jsxs(Fragment, { children: [
      buttonIcon && /* @__PURE__ */ jsx(
        "span",
        {
          "data-testid": "ockFundButtonIcon",
          className: "flex h-6 items-center",
          children: buttonIcon
        }
      ),
      hideText || /* @__PURE__ */ jsx("span", { "data-testid": "ockFundButtonTextContent", children: buttonTextContent })
    ] });
  }, [buttonState, buttonIcon, buttonTextContent, hideText]);
  if (openIn === "tab") {
    return /* @__PURE__ */ jsx(
      "a",
      {
        className: classNames,
        href: fundingUrlToRender,
        target: target ?? "_blank",
        rel,
        children: buttonContent
      }
    );
  }
  if (shouldShowConnectWallet) {
    return /* @__PURE__ */ jsx(ConnectWallet, { className: cn("w-full", className) });
  }
  return /* @__PURE__ */ jsx(
    "button",
    {
      className: classNames,
      onClick: handleClick,
      type: "button",
      disabled: isDisabled,
      "data-testid": "ockFundButton",
      children: buttonContent
    }
  );
}
export {
  FundButton
};
