// @coinbase/onchainkit/dist/fund/components/FundCardSubmitButton.js
'use client';
import { jsx } from "react/jsx-runtime";
import { useCallback, useMemo } from "react";
import { useFundCardFundingUrl } from "../hooks/useFundCardFundingUrl.js";
import { FundButton } from "./FundButton.js";
import { useFundContext } from "./FundCardProvider.js";
function FundCardSubmitButton() {
  const {
    fundAmountFiat,
    fundAmountCrypto,
    submitButtonState,
    setSubmitButtonState,
    buttonText,
    currency,
    updateLifecycleStatus
  } = useFundContext();
  const fundingUrl = useFundCardFundingUrl();
  const handleOnClick = useCallback(
    () => setSubmitButtonState("loading"),
    [setSubmitButtonState]
  );
  const handleOnPopupClose = useCallback(() => {
    updateLifecycleStatus({ statusName: "exit", statusData: void 0 });
    setSubmitButtonState("default");
  }, [updateLifecycleStatus, setSubmitButtonState]);
  const isButtonDisabled = useMemo(
    () => (!fundAmountFiat || Number(fundAmountCrypto) === 0) && (!fundAmountCrypto || Number(fundAmountFiat) === 0),
    [fundAmountCrypto, fundAmountFiat]
  );
  return /* @__PURE__ */ jsx(
    FundButton,
    {
      disabled: isButtonDisabled,
      hideIcon: submitButtonState === "default",
      text: buttonText,
      className: "w-full",
      fundingUrl,
      state: submitButtonState,
      onClick: handleOnClick,
      onPopupClose: handleOnPopupClose,
      fiatCurrency: currency
    }
  );
}
export {
  FundCardSubmitButton
};
