// @coinbase/onchainkit/dist/internal/components/amount-input/AmountInputTypeSwitch.js
import { jsx, jsxs } from "react/jsx-runtime";
import { cn, text, pressable } from "../../../styles/theme.js";
import { useCallback, useMemo } from "react";
import { useIcon } from "../../hooks/useIcon.js";
import { formatFiatAmount } from "../../utils/formatFiatAmount.js";
import { truncateDecimalPlaces } from "../../utils/truncateDecimalPlaces.js";
import { Skeleton } from "../Skeleton.js";
function AmountInputTypeSwitch({
  selectedInputType,
  setSelectedInputType,
  asset,
  fiatAmount,
  cryptoAmount,
  exchangeRate,
  exchangeRateLoading,
  currency,
  loadingDisplay = /* @__PURE__ */ jsx(Skeleton, { className: "h-[1.625rem]" }),
  className
}) {
  const iconSvg = useIcon({ icon: "toggle" });
  const handleToggle = useCallback(() => {
    setSelectedInputType(selectedInputType === "fiat" ? "crypto" : "fiat");
  }, [selectedInputType, setSelectedInputType]);
  const formatCrypto = useCallback(
    (amount) => {
      return `${truncateDecimalPlaces(amount || "0", 8)} ${asset}`;
    },
    [asset]
  );
  const amountLine = useMemo(() => {
    return /* @__PURE__ */ jsx("span", { "data-testid": "ockAmountLine", className: cn(text.label1), children: selectedInputType === "fiat" ? formatCrypto(cryptoAmount) : formatFiatAmount({
      amount: fiatAmount,
      currency,
      minimumFractionDigits: 0
    }) });
  }, [cryptoAmount, fiatAmount, selectedInputType, formatCrypto, currency]);
  if (exchangeRateLoading || !exchangeRate) {
    return loadingDisplay;
  }
  return /* @__PURE__ */ jsxs("div", { className: cn("flex items-center", className), children: [
    /* @__PURE__ */ jsx(
      "button",
      {
        type: "button",
        "aria-label": "amount type switch",
        className: cn(
          pressable.default,
          "mr-1 rounded-full p-1 opacity-50 transition-opacity hover:opacity-100"
        ),
        "data-testid": "ockAmountTypeSwitch",
        onClick: handleToggle,
        children: /* @__PURE__ */ jsx("div", { className: "h-[1.125rem] w-[1.125rem]", children: iconSvg })
      }
    ),
    /* @__PURE__ */ jsx("div", { className: "w-full truncate", children: amountLine })
  ] });
}
export {
  AmountInputTypeSwitch
};
